<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.3.450">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Dave Bridges">

<title>WGCNA Analysis of DO Livers</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="Liver-WGCNA-Construction_files/libs/clipboard/clipboard.min.js"></script>
<script src="Liver-WGCNA-Construction_files/libs/quarto-html/quarto.js"></script>
<script src="Liver-WGCNA-Construction_files/libs/quarto-html/popper.min.js"></script>
<script src="Liver-WGCNA-Construction_files/libs/quarto-html/tippy.umd.min.js"></script>
<script src="Liver-WGCNA-Construction_files/libs/quarto-html/anchor.min.js"></script>
<link href="Liver-WGCNA-Construction_files/libs/quarto-html/tippy.css" rel="stylesheet">
<link href="Liver-WGCNA-Construction_files/libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="Liver-WGCNA-Construction_files/libs/bootstrap/bootstrap.min.js"></script>
<link href="Liver-WGCNA-Construction_files/libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="Liver-WGCNA-Construction_files/libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">

  <script src="https://polyfill.io/v3/polyfill.min.js?features=es6"></script>
  <script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-chtml-full.js" type="text/javascript"></script>

</head>

<body class="fullcontent">

<div id="quarto-content" class="page-columns page-rows-contents page-layout-article">

<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">WGCNA Analysis of DO Livers</h1>
<p class="subtitle lead">Construction of Modules</p>
</div>



<div class="quarto-title-meta">

    <div>
    <div class="quarto-title-meta-heading">Author</div>
    <div class="quarto-title-meta-contents">
             <p>Dave Bridges </p>
          </div>
  </div>
    
  
    
  </div>
  

</header>

<p>Followed instructions on https://horvath.genetics.ucla.edu/html/CoexpressionNetwork/Rpackages/WGCNA/Tutorials/, which was based on the method used in https://doi.org/10.1371/journal.pgen.0020130</p>
<section id="data-preparation" class="level1">
<h1>Data Preparation</h1>
<div class="cell">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(WGCNA)</span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="fu">options</span>(<span class="at">stringsAsFactors =</span> F)</span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a>phenotype.datatfile <span class="ot">&lt;-</span> <span class="st">'../dataset.phenotype.Svenson_DO_HFD.v12.Rds'</span></span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a>all.phenotype.data <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(phenotype.datatfile)</span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a>annotation <span class="ot">&lt;-</span> all.phenotype.data<span class="sc">$</span>annot.samples[<span class="fu">c</span>(<span class="st">'mouse.id'</span>,<span class="st">'sex'</span>,<span class="st">'diet'</span>)]</span>
<span id="cb1-7"><a href="#cb1-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-8"><a href="#cb1-8" aria-hidden="true" tabindex="-1"></a>phenotype.data <span class="ot">&lt;-</span> </span>
<span id="cb1-9"><a href="#cb1-9" aria-hidden="true" tabindex="-1"></a>  all.phenotype.data<span class="sc">$</span>data<span class="sc">$</span>raw <span class="sc">%&gt;%</span> </span>
<span id="cb1-10"><a href="#cb1-10" aria-hidden="true" tabindex="-1"></a>  as.data.frame </span>
<span id="cb1-11"><a href="#cb1-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-12"><a href="#cb1-12" aria-hidden="true" tabindex="-1"></a>phenotype.data<span class="sc">$</span>mouse.id <span class="ot">&lt;-</span> <span class="fu">rownames</span>(phenotype.data)</span>
<span id="cb1-13"><a href="#cb1-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-14"><a href="#cb1-14" aria-hidden="true" tabindex="-1"></a>phenotype.data <span class="ot">&lt;-</span></span>
<span id="cb1-15"><a href="#cb1-15" aria-hidden="true" tabindex="-1"></a>  phenotype.data <span class="sc">%&gt;%</span></span>
<span id="cb1-16"><a href="#cb1-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(annotation,<span class="at">by=</span><span class="st">'mouse.id'</span>)</span>
<span id="cb1-17"><a href="#cb1-17" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb1-18"><a href="#cb1-18" aria-hidden="true" tabindex="-1"></a>expression.datatfile <span class="ot">&lt;-</span> <span class="st">'../dataset.mrna.Svenson_DO_HFD.v12.Rds'</span></span>
<span id="cb1-19"><a href="#cb1-19" aria-hidden="true" tabindex="-1"></a>expression.data <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(expression.datatfile)</span>
<span id="cb1-20"><a href="#cb1-20" aria-hidden="true" tabindex="-1"></a>mrna.data <span class="ot">&lt;-</span></span>
<span id="cb1-21"><a href="#cb1-21" aria-hidden="true" tabindex="-1"></a>  expression.data<span class="sc">$</span>data<span class="sc">$</span>raw <span class="sc">%&gt;%</span> as.data.frame</span>
<span id="cb1-22"><a href="#cb1-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-23"><a href="#cb1-23" aria-hidden="true" tabindex="-1"></a>good.genes <span class="ot">&lt;-</span> <span class="fu">goodSamplesGenes</span>(mrna.data)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code> Flagging genes and samples with too many missing values...
  ..step 1</code></pre>
</div>
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="co">#check if all genes are ok, not too many missing values for genes or samples</span></span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a>good.genes<span class="sc">$</span>allOK</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] TRUE</code></pre>
</div>
</div>
</section>
<section id="network-construction" class="level1">
<h1>Network Construction</h1>
<div class="cell">
<div class="sourceCode cell-code" id="cb5"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a>nSets <span class="ot">&lt;-</span> <span class="dv">4</span> <span class="co">#four datasets, NCD and HFD, male and female</span></span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a>setLabels <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"NCD-M"</span>,<span class="st">"HFHS-M"</span>,<span class="st">"NCD-F"</span>,<span class="st">"HFHS-F"</span>)</span>
<span id="cb5-3"><a href="#cb5-3" aria-hidden="true" tabindex="-1"></a>multiExpr<span class="ot">=</span><span class="fu">vector</span>(<span class="at">mode=</span><span class="st">"list"</span>,<span class="at">length=</span>nSets) <span class="co">#create vector with two datasets</span></span>
<span id="cb5-4"><a href="#cb5-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-5"><a href="#cb5-5" aria-hidden="true" tabindex="-1"></a><span class="co">#filter for only exprerssion data where we have appropriatet phenotype data</span></span>
<span id="cb5-6"><a href="#cb5-6" aria-hidden="true" tabindex="-1"></a>chow.mice.m <span class="ot">&lt;-</span> phenotype.data <span class="sc">%&gt;%</span> <span class="fu">filter</span>(diet<span class="sc">==</span><span class="st">"chow"</span><span class="sc">&amp;</span>sex<span class="sc">==</span><span class="st">"M"</span>) <span class="sc">%&gt;%</span> <span class="fu">pull</span>(mouse.id)</span>
<span id="cb5-7"><a href="#cb5-7" aria-hidden="true" tabindex="-1"></a>hf.mice.m <span class="ot">&lt;-</span> phenotype.data <span class="sc">%&gt;%</span> <span class="fu">filter</span>(diet<span class="sc">==</span><span class="st">"hf"</span><span class="sc">&amp;</span>sex<span class="sc">==</span><span class="st">"M"</span>) <span class="sc">%&gt;%</span> <span class="fu">pull</span>(mouse.id)</span>
<span id="cb5-8"><a href="#cb5-8" aria-hidden="true" tabindex="-1"></a>chow.mice.f <span class="ot">&lt;-</span> phenotype.data <span class="sc">%&gt;%</span> <span class="fu">filter</span>(diet<span class="sc">==</span><span class="st">"chow"</span><span class="sc">&amp;</span>sex<span class="sc">==</span><span class="st">"F"</span>) <span class="sc">%&gt;%</span> <span class="fu">pull</span>(mouse.id)</span>
<span id="cb5-9"><a href="#cb5-9" aria-hidden="true" tabindex="-1"></a>hf.mice.f <span class="ot">&lt;-</span> phenotype.data <span class="sc">%&gt;%</span> <span class="fu">filter</span>(diet<span class="sc">==</span><span class="st">"hf"</span><span class="sc">&amp;</span>sex<span class="sc">==</span><span class="st">"F"</span>) <span class="sc">%&gt;%</span> <span class="fu">pull</span>(mouse.id)</span>
<span id="cb5-10"><a href="#cb5-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-11"><a href="#cb5-11" aria-hidden="true" tabindex="-1"></a>chow.expression.m <span class="ot">&lt;-</span> mrna.data[chow.mice.m,] <span class="sc">%&gt;%</span> na.omit</span>
<span id="cb5-12"><a href="#cb5-12" aria-hidden="true" tabindex="-1"></a>hf.expression.m <span class="ot">&lt;-</span> mrna.data[hf.mice.m,] <span class="sc">%&gt;%</span> na.omit</span>
<span id="cb5-13"><a href="#cb5-13" aria-hidden="true" tabindex="-1"></a>chow.expression.f <span class="ot">&lt;-</span> mrna.data[chow.mice.f,] <span class="sc">%&gt;%</span> na.omit</span>
<span id="cb5-14"><a href="#cb5-14" aria-hidden="true" tabindex="-1"></a>hf.expression.f <span class="ot">&lt;-</span> mrna.data[hf.mice.f,] <span class="sc">%&gt;%</span> na.omit</span>
<span id="cb5-15"><a href="#cb5-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-16"><a href="#cb5-16" aria-hidden="true" tabindex="-1"></a><span class="co"># enter data on expression in to multiExpr object</span></span>
<span id="cb5-17"><a href="#cb5-17" aria-hidden="true" tabindex="-1"></a>multiExpr[[<span class="dv">1</span>]]<span class="ot">=</span><span class="fu">list</span>(<span class="at">data=</span>chow.expression.m)</span>
<span id="cb5-18"><a href="#cb5-18" aria-hidden="true" tabindex="-1"></a>multiExpr[[<span class="dv">2</span>]]<span class="ot">=</span><span class="fu">list</span>(<span class="at">data=</span>hf.expression.m) </span>
<span id="cb5-19"><a href="#cb5-19" aria-hidden="true" tabindex="-1"></a>multiExpr[[<span class="dv">3</span>]]<span class="ot">=</span><span class="fu">list</span>(<span class="at">data=</span>chow.expression.f)</span>
<span id="cb5-20"><a href="#cb5-20" aria-hidden="true" tabindex="-1"></a>multiExpr[[<span class="dv">4</span>]]<span class="ot">=</span><span class="fu">list</span>(<span class="at">data=</span>hf.expression.f) </span>
<span id="cb5-21"><a href="#cb5-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-22"><a href="#cb5-22" aria-hidden="true" tabindex="-1"></a>exprSize <span class="ot">&lt;-</span> <span class="fu">checkSets</span>(multiExpr)</span>
<span id="cb5-23"><a href="#cb5-23" aria-hidden="true" tabindex="-1"></a>gsg <span class="ot">&lt;-</span> <span class="fu">goodSamplesGenesMS</span>(multiExpr)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code> Flagging genes and samples with too many missing values...
  ..step 1</code></pre>
</div>
<div class="sourceCode cell-code" id="cb7"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a><span class="co"># remove genes with too many missing values or no variance</span></span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span>gsg<span class="sc">$</span>allOK)</span>
<span id="cb7-3"><a href="#cb7-3" aria-hidden="true" tabindex="-1"></a>{</span>
<span id="cb7-4"><a href="#cb7-4" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Print information about the removed genes:</span></span>
<span id="cb7-5"><a href="#cb7-5" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (<span class="fu">sum</span>(<span class="sc">!</span>gsg<span class="sc">$</span>goodGenes) <span class="sc">&gt;</span> <span class="dv">0</span>)</span>
<span id="cb7-6"><a href="#cb7-6" aria-hidden="true" tabindex="-1"></a>    <span class="fu">printFlush</span>(<span class="fu">paste</span>(<span class="st">"Removing genes:"</span>, <span class="fu">paste</span>(<span class="fu">names</span>(multiExpr[[<span class="dv">1</span>]]<span class="sc">$</span>data)[<span class="sc">!</span>gsg<span class="sc">$</span>goodGenes], </span>
<span id="cb7-7"><a href="#cb7-7" aria-hidden="true" tabindex="-1"></a>                                              <span class="at">collapse =</span> <span class="st">", "</span>)))</span>
<span id="cb7-8"><a href="#cb7-8" aria-hidden="true" tabindex="-1"></a>  <span class="cf">for</span> (set <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span>exprSize<span class="sc">$</span>nSets)</span>
<span id="cb7-9"><a href="#cb7-9" aria-hidden="true" tabindex="-1"></a>  {</span>
<span id="cb7-10"><a href="#cb7-10" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> (<span class="fu">sum</span>(<span class="sc">!</span>gsg<span class="sc">$</span>goodSamples[[set]]))</span>
<span id="cb7-11"><a href="#cb7-11" aria-hidden="true" tabindex="-1"></a>      <span class="fu">printFlush</span>(<span class="fu">paste</span>(<span class="st">"In set"</span>, setLabels[set], <span class="st">"removing samples"</span>,</span>
<span id="cb7-12"><a href="#cb7-12" aria-hidden="true" tabindex="-1"></a>                       <span class="fu">paste</span>(<span class="fu">rownames</span>(multiExpr[[set]]<span class="sc">$</span>data)[<span class="sc">!</span>gsg<span class="sc">$</span>goodSamples[[set]]], <span class="at">collapse =</span> <span class="st">", "</span>)))</span>
<span id="cb7-13"><a href="#cb7-13" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Remove the offending genes and samples</span></span>
<span id="cb7-14"><a href="#cb7-14" aria-hidden="true" tabindex="-1"></a>    multiExpr[[set]]<span class="sc">$</span>data <span class="ot">=</span> multiExpr[[set]]<span class="sc">$</span>data[gsg<span class="sc">$</span>goodSamples[[set]], gsg<span class="sc">$</span>goodGenes];</span>
<span id="cb7-15"><a href="#cb7-15" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb7-16"><a href="#cb7-16" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Update exprSize</span></span>
<span id="cb7-17"><a href="#cb7-17" aria-hidden="true" tabindex="-1"></a>  exprSize <span class="ot">=</span> <span class="fu">checkSets</span>(multiExpr)</span>
<span id="cb7-18"><a href="#cb7-18" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<section id="clustering-expression-datasets" class="level2">
<h2 class="anchored" data-anchor-id="clustering-expression-datasets">Clustering Expression Datasets</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb8"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a><span class="co">#crerate two sample trees</span></span>
<span id="cb8-2"><a href="#cb8-2" aria-hidden="true" tabindex="-1"></a>sampleTrees <span class="ot">=</span> <span class="fu">list</span>()</span>
<span id="cb8-3"><a href="#cb8-3" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (set <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span>nSets)</span>
<span id="cb8-4"><a href="#cb8-4" aria-hidden="true" tabindex="-1"></a>{</span>
<span id="cb8-5"><a href="#cb8-5" aria-hidden="true" tabindex="-1"></a>  sampleTrees[[set]] <span class="ot">=</span> <span class="fu">hclust</span>(<span class="fu">dist</span>(multiExpr[[set]]<span class="sc">$</span>data), <span class="at">method =</span> <span class="st">"average"</span>)</span>
<span id="cb8-6"><a href="#cb8-6" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb8-7"><a href="#cb8-7" aria-hidden="true" tabindex="-1"></a><span class="co">#cluster genes in each set, make a hclust tree using "average" method</span></span>
<span id="cb8-8"><a href="#cb8-8" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (set <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span>nSets)</span>
<span id="cb8-9"><a href="#cb8-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">plot</span>(sampleTrees[[set]], </span>
<span id="cb8-10"><a href="#cb8-10" aria-hidden="true" tabindex="-1"></a>       <span class="at">main =</span> <span class="fu">paste</span>(<span class="st">"Sample clustering on all genes in"</span>,</span>
<span id="cb8-11"><a href="#cb8-11" aria-hidden="true" tabindex="-1"></a>                    setLabels[set]),</span>
<span id="cb8-12"><a href="#cb8-12" aria-hidden="true" tabindex="-1"></a>       <span class="at">xlab=</span><span class="st">""</span>, </span>
<span id="cb8-13"><a href="#cb8-13" aria-hidden="true" tabindex="-1"></a>       <span class="at">sub=</span><span class="st">""</span>, </span>
<span id="cb8-14"><a href="#cb8-14" aria-hidden="true" tabindex="-1"></a>       <span class="at">cex =</span> <span class="fl">0.7</span>);</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="figures/diet-clustered-trees-1.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output-display">
<p><img src="figures/diet-clustered-trees-2.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output-display">
<p><img src="figures/diet-clustered-trees-3.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output-display">
<p><img src="figures/diet-clustered-trees-4.png" class="img-fluid" width="672"></p>
</div>
<div class="sourceCode cell-code" id="cb9"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Form a multi-set structure called Traits that will hold the clinical traits.</span></span>
<span id="cb9-2"><a href="#cb9-2" aria-hidden="true" tabindex="-1"></a>Traits <span class="ot">=</span> <span class="fu">vector</span>(<span class="at">mode=</span><span class="st">"list"</span>, <span class="at">length =</span> nSets);</span>
<span id="cb9-3"><a href="#cb9-3" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (set <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span>nSets)</span>
<span id="cb9-4"><a href="#cb9-4" aria-hidden="true" tabindex="-1"></a>{</span>
<span id="cb9-5"><a href="#cb9-5" aria-hidden="true" tabindex="-1"></a>  setSamples <span class="ot">=</span> <span class="fu">rownames</span>(multiExpr[[set]]<span class="sc">$</span>data);</span>
<span id="cb9-6"><a href="#cb9-6" aria-hidden="true" tabindex="-1"></a>  traitRows <span class="ot">=</span> <span class="fu">match</span>(setSamples, <span class="fu">rownames</span>(phenotype.data));</span>
<span id="cb9-7"><a href="#cb9-7" aria-hidden="true" tabindex="-1"></a>  Traits[[set]] <span class="ot">=</span> <span class="fu">list</span>(<span class="at">data =</span> phenotype.data[traitRows,]);</span>
<span id="cb9-8"><a href="#cb9-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rownames</span>(Traits[[set]]<span class="sc">$</span>data) <span class="ot">=</span> <span class="fu">rownames</span>(phenotype.data[traitRows,]);</span>
<span id="cb9-9"><a href="#cb9-9" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb9-10"><a href="#cb9-10" aria-hidden="true" tabindex="-1"></a><span class="fu">collectGarbage</span>();</span>
<span id="cb9-11"><a href="#cb9-11" aria-hidden="true" tabindex="-1"></a><span class="co"># Define data set dimensions</span></span>
<span id="cb9-12"><a href="#cb9-12" aria-hidden="true" tabindex="-1"></a>nGenes <span class="ot">=</span> exprSize<span class="sc">$</span>nGenes;</span>
<span id="cb9-13"><a href="#cb9-13" aria-hidden="true" tabindex="-1"></a>nSamples <span class="ot">=</span> exprSize<span class="sc">$</span>nSamples;</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>The combined dataset includes 11770 genes and 120112123120 samples separated into <code>nSets</code> datasets (NCD-M,HFHS-M,NCD-F,HFHS-F).</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb10"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Choose a set of soft-thresholding powers</span></span>
<span id="cb10-2"><a href="#cb10-2" aria-hidden="true" tabindex="-1"></a>powers <span class="ot">=</span> <span class="fu">seq</span>(<span class="dv">4</span>,<span class="dv">20</span>,<span class="at">by=</span><span class="dv">1</span>)</span>
<span id="cb10-3"><a href="#cb10-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Initialize a list to hold the results of scale-free analysis</span></span>
<span id="cb10-4"><a href="#cb10-4" aria-hidden="true" tabindex="-1"></a>powerTables <span class="ot">=</span> <span class="fu">vector</span>(<span class="at">mode =</span> <span class="st">"list"</span>, <span class="at">length =</span> nSets);</span>
<span id="cb10-5"><a href="#cb10-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Call the network topology analysis function for each set in turn</span></span>
<span id="cb10-6"><a href="#cb10-6" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (set <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span>nSets)</span>
<span id="cb10-7"><a href="#cb10-7" aria-hidden="true" tabindex="-1"></a>  powerTables[[set]] <span class="ot">=</span> <span class="fu">list</span>(<span class="at">data =</span> <span class="fu">pickSoftThreshold</span>(multiExpr[[set]]<span class="sc">$</span>data,</span>
<span id="cb10-8"><a href="#cb10-8" aria-hidden="true" tabindex="-1"></a>                                                     <span class="at">powerVector=</span>powers,</span>
<span id="cb10-9"><a href="#cb10-9" aria-hidden="true" tabindex="-1"></a>                                                     <span class="at">verbose =</span> <span class="dv">2</span>)[[<span class="dv">2</span>]]);</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>pickSoftThreshold: will use block size 3801.
 pickSoftThreshold: calculating connectivity for given powers...
   ..working on genes 1 through 3801 of 11770</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>   ..working on genes 3802 through 7602 of 11770
   ..working on genes 7603 through 11403 of 11770
   ..working on genes 11404 through 11770 of 11770
   Power SFT.R.sq  slope truncated.R.sq mean.k. median.k. max.k.
1      4    0.316 -0.365          0.223  1640.0    1670.0   3930
2      5    0.462 -0.532          0.467  1180.0    1120.0   3250
3      6    0.538 -0.674          0.607   879.0     771.0   2720
4      7    0.578 -0.788          0.689   667.0     534.0   2310
5      8    0.607 -0.890          0.746   515.0     375.0   1980
6      9    0.636 -0.971          0.796   403.0     265.0   1710
7     10    0.647 -1.040          0.829   320.0     190.0   1480
8     11    0.668 -1.110          0.854   256.0     137.0   1290
9     12    0.677 -1.180          0.872   207.0      99.2   1140
10    13    0.690 -1.230          0.891   169.0      72.2   1000
11    14    0.696 -1.270          0.904   139.0      53.0    886
12    15    0.711 -1.310          0.916   115.0      39.2    786
13    16    0.718 -1.340          0.926    95.4      29.2    700
14    17    0.735 -1.380          0.941    79.8      21.8    626
15    18    0.746 -1.400          0.951    67.1      16.4    561
16    19    0.738 -1.450          0.948    56.7      12.3    503
17    20    0.742 -1.490          0.950    48.1       9.3    453
pickSoftThreshold: will use block size 3801.
 pickSoftThreshold: calculating connectivity for given powers...
   ..working on genes 1 through 3801 of 11770
   ..working on genes 3802 through 7602 of 11770
   ..working on genes 7603 through 11403 of 11770
   ..working on genes 11404 through 11770 of 11770
   Power SFT.R.sq  slope truncated.R.sq mean.k. median.k. max.k.
1      4    0.328 -0.701          0.519  1090.0   1080.00   2940
2      5    0.425 -0.882          0.642   743.0    694.00   2300
3      6    0.482 -1.030          0.721   523.0    453.00   1830
4      7    0.520 -1.160          0.767   379.0    302.00   1480
5      8    0.561 -1.230          0.808   281.0    205.00   1220
6      9    0.579 -1.300          0.835   213.0    141.00   1010
7     10    0.601 -1.350          0.859   164.0     98.80    840
8     11    0.626 -1.380          0.880   128.0     69.70    707
9     12    0.647 -1.410          0.896   101.0     49.80    599
10    13    0.657 -1.440          0.906    80.3     35.80    511
11    14    0.667 -1.470          0.914    64.6     25.90    439
12    15    0.664 -1.520          0.913    52.5     19.00    379
13    16    0.665 -1.560          0.917    42.9     14.00    329
14    17    0.678 -1.570          0.925    35.4     10.40    287
15    18    0.688 -1.580          0.932    29.3      7.71    251
16    19    0.682 -1.620          0.931    24.5      5.82    221
17    20    0.693 -1.640          0.939    20.5      4.43    195
pickSoftThreshold: will use block size 3801.
 pickSoftThreshold: calculating connectivity for given powers...
   ..working on genes 1 through 3801 of 11770
   ..working on genes 3802 through 7602 of 11770
   ..working on genes 7603 through 11403 of 11770
   ..working on genes 11404 through 11770 of 11770
   Power SFT.R.sq  slope truncated.R.sq mean.k. median.k. max.k.
1      4    0.344 -0.349          0.227  1660.0    1690.0   3960
2      5    0.484 -0.517          0.456  1210.0    1150.0   3280
3      6    0.559 -0.645          0.582   901.0     789.0   2750
4      7    0.614 -0.740          0.683   687.0     549.0   2340
5      8    0.653 -0.819          0.744   534.0     387.0   2000
6      9    0.697 -0.876          0.804   421.0     276.0   1730
7     10    0.724 -0.932          0.844   336.0     198.0   1500
8     11    0.747 -0.972          0.872   271.0     143.0   1310
9     12    0.759 -1.020          0.887   220.0     105.0   1150
10    13    0.771 -1.070          0.903   181.0      76.5   1010
11    14    0.776 -1.110          0.915   150.0      56.4    900
12    15    0.774 -1.160          0.916   125.0      41.7    803
13    16    0.772 -1.200          0.918   104.0      31.1    720
14    17    0.773 -1.240          0.924    88.0      23.2    647
15    18    0.779 -1.270          0.931    74.5      17.5    583
16    19    0.781 -1.300          0.934    63.4      13.2    527
17    20    0.766 -1.340          0.924    54.2      10.1    478
pickSoftThreshold: will use block size 3801.
 pickSoftThreshold: calculating connectivity for given powers...
   ..working on genes 1 through 3801 of 11770
   ..working on genes 3802 through 7602 of 11770
   ..working on genes 7603 through 11403 of 11770
   ..working on genes 11404 through 11770 of 11770
   Power SFT.R.sq  slope truncated.R.sq mean.k. median.k. max.k.
1      4    0.243 -0.452          0.306  1560.0    1600.0   3760
2      5    0.360 -0.677          0.490  1110.0    1080.0   3080
3      6    0.424 -0.843          0.597   816.0     750.0   2550
4      7    0.468 -0.977          0.666   614.0     525.0   2140
5      8    0.513 -1.070          0.726   472.0     373.0   1820
6      9    0.538 -1.160          0.764   368.0     269.0   1550
7     10    0.568 -1.230          0.804   291.0     195.0   1340
8     11    0.590 -1.270          0.831   233.0     143.0   1160
9     12    0.604 -1.330          0.847   188.0     106.0   1010
10    13    0.618 -1.370          0.864   154.0      78.2    882
11    14    0.638 -1.400          0.882   126.0      58.8    774
12    15    0.647 -1.440          0.892   105.0      44.4    683
13    16    0.663 -1.460          0.905    87.5      33.6    604
14    17    0.675 -1.470          0.915    73.4      25.5    536
15    18    0.688 -1.490          0.924    62.0      19.5    477
16    19    0.691 -1.510          0.927    52.6      15.1    426
17    20    0.702 -1.520          0.935    44.9      11.7    381</code></pre>
</div>
<div class="sourceCode cell-code" id="cb13"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a><span class="fu">collectGarbage</span>();</span>
<span id="cb13-2"><a href="#cb13-2" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot the results:</span></span>
<span id="cb13-3"><a href="#cb13-3" aria-hidden="true" tabindex="-1"></a>colors <span class="ot">=</span> color.scheme</span>
<span id="cb13-4"><a href="#cb13-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-5"><a href="#cb13-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Will plot these columns of the returned scale free analysis tables</span></span>
<span id="cb13-6"><a href="#cb13-6" aria-hidden="true" tabindex="-1"></a>plotCols <span class="ot">=</span> <span class="fu">c</span>(<span class="dv">2</span>,<span class="dv">5</span>,<span class="dv">6</span>,<span class="dv">7</span>)</span>
<span id="cb13-7"><a href="#cb13-7" aria-hidden="true" tabindex="-1"></a>colNames <span class="ot">=</span> <span class="fu">c</span>(<span class="st">"Scale Free Topology Model Fit"</span>, <span class="st">"Mean connectivity"</span>, <span class="st">"Median connectivity"</span>,</span>
<span id="cb13-8"><a href="#cb13-8" aria-hidden="true" tabindex="-1"></a><span class="st">"Max connectivity"</span>);</span>
<span id="cb13-9"><a href="#cb13-9" aria-hidden="true" tabindex="-1"></a><span class="co"># Get the minima and maxima of the plotted points</span></span>
<span id="cb13-10"><a href="#cb13-10" aria-hidden="true" tabindex="-1"></a>ylim <span class="ot">=</span> <span class="fu">matrix</span>(<span class="cn">NA</span>, <span class="at">nrow =</span> <span class="dv">2</span>, <span class="at">ncol =</span> <span class="dv">4</span>);</span>
<span id="cb13-11"><a href="#cb13-11" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (set <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span>nSets)</span>
<span id="cb13-12"><a href="#cb13-12" aria-hidden="true" tabindex="-1"></a>{</span>
<span id="cb13-13"><a href="#cb13-13" aria-hidden="true" tabindex="-1"></a>  <span class="cf">for</span> (col <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(plotCols))</span>
<span id="cb13-14"><a href="#cb13-14" aria-hidden="true" tabindex="-1"></a>  {</span>
<span id="cb13-15"><a href="#cb13-15" aria-hidden="true" tabindex="-1"></a>    ylim[<span class="dv">1</span>, col] <span class="ot">=</span> <span class="fu">min</span>(ylim[<span class="dv">1</span>, col], powerTables[[set]]<span class="sc">$</span>data[, plotCols[col]], <span class="at">na.rm =</span> <span class="cn">TRUE</span>);</span>
<span id="cb13-16"><a href="#cb13-16" aria-hidden="true" tabindex="-1"></a>    ylim[<span class="dv">2</span>, col] <span class="ot">=</span> <span class="fu">max</span>(ylim[<span class="dv">2</span>, col], powerTables[[set]]<span class="sc">$</span>data[, plotCols[col]], <span class="at">na.rm =</span> <span class="cn">TRUE</span>);</span>
<span id="cb13-17"><a href="#cb13-17" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb13-18"><a href="#cb13-18" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb13-19"><a href="#cb13-19" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot the quantities in the chosen columns vs. the soft thresholding power</span></span>
<span id="cb13-20"><a href="#cb13-20" aria-hidden="true" tabindex="-1"></a><span class="fu">sizeGrWindow</span>(<span class="dv">8</span>, <span class="dv">6</span>)</span>
<span id="cb13-21"><a href="#cb13-21" aria-hidden="true" tabindex="-1"></a><span class="fu">pdf</span>(<span class="at">file =</span> <span class="st">"scaleFreeAnalysis.pdf"</span>, <span class="at">wi =</span> <span class="dv">8</span>, <span class="at">he =</span> <span class="dv">6</span>)</span>
<span id="cb13-22"><a href="#cb13-22" aria-hidden="true" tabindex="-1"></a><span class="fu">par</span>(<span class="at">mfcol =</span> <span class="fu">c</span>(<span class="dv">2</span>,<span class="dv">2</span>));</span>
<span id="cb13-23"><a href="#cb13-23" aria-hidden="true" tabindex="-1"></a><span class="fu">par</span>(<span class="at">mar =</span> <span class="fu">c</span>(<span class="fl">4.2</span>, <span class="fl">4.2</span> , <span class="fl">2.2</span>, <span class="fl">0.5</span>))</span>
<span id="cb13-24"><a href="#cb13-24" aria-hidden="true" tabindex="-1"></a>cex1 <span class="ot">=</span> <span class="fl">0.7</span>;</span>
<span id="cb13-25"><a href="#cb13-25" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (col <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(plotCols)) <span class="cf">for</span> (set <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span>nSets)</span>
<span id="cb13-26"><a href="#cb13-26" aria-hidden="true" tabindex="-1"></a>{</span>
<span id="cb13-27"><a href="#cb13-27" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (set<span class="sc">==</span><span class="dv">1</span>)</span>
<span id="cb13-28"><a href="#cb13-28" aria-hidden="true" tabindex="-1"></a>  {</span>
<span id="cb13-29"><a href="#cb13-29" aria-hidden="true" tabindex="-1"></a>    <span class="fu">plot</span>(powerTables[[set]]<span class="sc">$</span>data[,<span class="dv">1</span>], <span class="sc">-</span><span class="fu">sign</span>(powerTables[[set]]<span class="sc">$</span>data[,<span class="dv">3</span>])<span class="sc">*</span>powerTables[[set]]<span class="sc">$</span>data[,<span class="dv">2</span>],</span>
<span id="cb13-30"><a href="#cb13-30" aria-hidden="true" tabindex="-1"></a>         <span class="at">xlab=</span><span class="st">"Soft Threshold (power)"</span>,<span class="at">ylab=</span>colNames[col],<span class="at">type=</span><span class="st">"n"</span>, <span class="at">ylim =</span> ylim[, col],</span>
<span id="cb13-31"><a href="#cb13-31" aria-hidden="true" tabindex="-1"></a>         <span class="at">main =</span> colNames[col]);</span>
<span id="cb13-32"><a href="#cb13-32" aria-hidden="true" tabindex="-1"></a>    <span class="fu">addGrid</span>();</span>
<span id="cb13-33"><a href="#cb13-33" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb13-34"><a href="#cb13-34" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (col<span class="sc">==</span><span class="dv">1</span>)</span>
<span id="cb13-35"><a href="#cb13-35" aria-hidden="true" tabindex="-1"></a>  {</span>
<span id="cb13-36"><a href="#cb13-36" aria-hidden="true" tabindex="-1"></a>    <span class="fu">text</span>(powerTables[[set]]<span class="sc">$</span>data[,<span class="dv">1</span>], <span class="sc">-</span><span class="fu">sign</span>(powerTables[[set]]<span class="sc">$</span>data[,<span class="dv">3</span>])<span class="sc">*</span>powerTables[[set]]<span class="sc">$</span>data[,<span class="dv">2</span>],</span>
<span id="cb13-37"><a href="#cb13-37" aria-hidden="true" tabindex="-1"></a>         <span class="at">labels=</span>powers,<span class="at">cex=</span>cex1,<span class="at">col=</span>colors[set]);</span>
<span id="cb13-38"><a href="#cb13-38" aria-hidden="true" tabindex="-1"></a>  } <span class="cf">else</span></span>
<span id="cb13-39"><a href="#cb13-39" aria-hidden="true" tabindex="-1"></a>    <span class="fu">text</span>(powerTables[[set]]<span class="sc">$</span>data[,<span class="dv">1</span>], powerTables[[set]]<span class="sc">$</span>data[,plotCols[col]],</span>
<span id="cb13-40"><a href="#cb13-40" aria-hidden="true" tabindex="-1"></a>         <span class="at">labels=</span>powers,<span class="at">cex=</span>cex1,<span class="at">col=</span>colors[set]);</span>
<span id="cb13-41"><a href="#cb13-41" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (col<span class="sc">==</span><span class="dv">1</span>)</span>
<span id="cb13-42"><a href="#cb13-42" aria-hidden="true" tabindex="-1"></a>  {</span>
<span id="cb13-43"><a href="#cb13-43" aria-hidden="true" tabindex="-1"></a>    <span class="fu">legend</span>(<span class="st">"bottomright"</span>, <span class="at">legend =</span> setLabels, <span class="at">col =</span> colors, <span class="at">pch =</span> <span class="dv">20</span>) ;</span>
<span id="cb13-44"><a href="#cb13-44" aria-hidden="true" tabindex="-1"></a>  } <span class="cf">else</span></span>
<span id="cb13-45"><a href="#cb13-45" aria-hidden="true" tabindex="-1"></a>    <span class="fu">legend</span>(<span class="st">"topright"</span>, <span class="at">legend =</span> setLabels, <span class="at">col =</span> colors, <span class="at">pch =</span> <span class="dv">20</span>) ;</span>
<span id="cb13-46"><a href="#cb13-46" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb13-47"><a href="#cb13-47" aria-hidden="true" tabindex="-1"></a><span class="fu">dev.off</span>();</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>pdf 
  2 </code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb15"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" aria-hidden="true" tabindex="-1"></a>network.power <span class="ot">&lt;-</span> <span class="dv">15</span> <span class="co">#based on the max SFT R2 maximum</span></span>
<span id="cb15-2"><a href="#cb15-2" aria-hidden="true" tabindex="-1"></a>net <span class="ot">&lt;-</span> <span class="fu">blockwiseConsensusModules</span>(</span>
<span id="cb15-3"><a href="#cb15-3" aria-hidden="true" tabindex="-1"></a>        multiExpr, </span>
<span id="cb15-4"><a href="#cb15-4" aria-hidden="true" tabindex="-1"></a>        <span class="at">power =</span> network.power, </span>
<span id="cb15-5"><a href="#cb15-5" aria-hidden="true" tabindex="-1"></a>        <span class="at">minModuleSize =</span> <span class="dv">30</span>, <span class="co">#default is 20, tutorial said 30</span></span>
<span id="cb15-6"><a href="#cb15-6" aria-hidden="true" tabindex="-1"></a>        <span class="at">deepSplit =</span> <span class="dv">2</span>, <span class="co">#default</span></span>
<span id="cb15-7"><a href="#cb15-7" aria-hidden="true" tabindex="-1"></a>        <span class="at">maxBlockSize =</span> <span class="fu">ncol</span>(mrna.data), <span class="co">#total number of genes</span></span>
<span id="cb15-8"><a href="#cb15-8" aria-hidden="true" tabindex="-1"></a>        <span class="at">pamRespectsDendro =</span> <span class="cn">FALSE</span>, </span>
<span id="cb15-9"><a href="#cb15-9" aria-hidden="true" tabindex="-1"></a>        <span class="at">mergeCutHeight =</span> <span class="fl">0.25</span>,</span>
<span id="cb15-10"><a href="#cb15-10" aria-hidden="true" tabindex="-1"></a>        <span class="at">numericLabels =</span> <span class="cn">TRUE</span>,</span>
<span id="cb15-11"><a href="#cb15-11" aria-hidden="true" tabindex="-1"></a>        <span class="at">minKMEtoStay =</span> <span class="fl">0.1</span>, <span class="co">#default is 0.2, tutorial said 0.1</span></span>
<span id="cb15-12"><a href="#cb15-12" aria-hidden="true" tabindex="-1"></a>        <span class="at">saveTOMs =</span> <span class="cn">TRUE</span>, </span>
<span id="cb15-13"><a href="#cb15-13" aria-hidden="true" tabindex="-1"></a>        <span class="at">verbose =</span> <span class="dv">1</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code> Calculating consensus modules and module eigengenes block-wise from all genes
 Calculating topological overlaps block-wise from all genes</code></pre>
</div>
</div>
<p>Constructed modules with 15 soft-thresholding power, based on the visualization of the soft-thresholding power <span class="math inline">\(R^2\)</span>.</p>
</section>
<section id="plotting-samples-clustered-by-consensus-modules" class="level2">
<h2 class="anchored" data-anchor-id="plotting-samples-clustered-by-consensus-modules">Plotting Samples Clustered by Consensus Modules</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb17"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1" aria-hidden="true" tabindex="-1"></a>consMEs <span class="ot">=</span> net<span class="sc">$</span>multiMEs</span>
<span id="cb17-2"><a href="#cb17-2" aria-hidden="true" tabindex="-1"></a>moduleLabels <span class="ot">=</span> net<span class="sc">$</span>colors</span>
<span id="cb17-3"><a href="#cb17-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Convert the numeric labels to color labels</span></span>
<span id="cb17-4"><a href="#cb17-4" aria-hidden="true" tabindex="-1"></a>moduleColors <span class="ot">=</span> <span class="fu">labels2colors</span>(moduleLabels)</span>
<span id="cb17-5"><a href="#cb17-5" aria-hidden="true" tabindex="-1"></a>consTree <span class="ot">=</span> net<span class="sc">$</span>dendrograms[[<span class="dv">1</span>]]</span>
<span id="cb17-6"><a href="#cb17-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-7"><a href="#cb17-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Convert labels to colors for plotting</span></span>
<span id="cb17-8"><a href="#cb17-8" aria-hidden="true" tabindex="-1"></a>mergedColors <span class="ot">=</span> <span class="fu">labels2colors</span>(net<span class="sc">$</span>colors)</span>
<span id="cb17-9"><a href="#cb17-9" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot the dendrogram and the module colors underneath</span></span>
<span id="cb17-10"><a href="#cb17-10" aria-hidden="true" tabindex="-1"></a><span class="fu">plotDendroAndColors</span>(net<span class="sc">$</span>dendrograms[[<span class="dv">1</span>]], mergedColors[net<span class="sc">$</span>blockGenes[[<span class="dv">1</span>]]],</span>
<span id="cb17-11"><a href="#cb17-11" aria-hidden="true" tabindex="-1"></a>                    <span class="st">"Module colors"</span>,</span>
<span id="cb17-12"><a href="#cb17-12" aria-hidden="true" tabindex="-1"></a>                    <span class="at">dendroLabels =</span> <span class="cn">FALSE</span>, <span class="at">hang =</span> <span class="fl">0.03</span>,</span>
<span id="cb17-13"><a href="#cb17-13" aria-hidden="true" tabindex="-1"></a>                    <span class="at">addGuide =</span> <span class="cn">TRUE</span>, <span class="at">guideHang =</span> <span class="fl">0.05</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="figures/wgcna-dendrograms-by-modules-1.png" class="img-fluid" width="672"></p>
</div>
<div class="sourceCode cell-code" id="cb18"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Convert labels to colors for plotting</span></span>
<span id="cb18-2"><a href="#cb18-2" aria-hidden="true" tabindex="-1"></a>mergedColors <span class="ot">=</span> <span class="fu">labels2colors</span>(net<span class="sc">$</span>colors)</span>
<span id="cb18-3"><a href="#cb18-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-4"><a href="#cb18-4" aria-hidden="true" tabindex="-1"></a>MEs <span class="ot">=</span> net<span class="sc">$</span>MEs;</span>
<span id="cb18-5"><a href="#cb18-5" aria-hidden="true" tabindex="-1"></a>geneTree <span class="ot">=</span> net<span class="sc">$</span>dendrograms[[<span class="dv">1</span>]];</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="output" class="level1">
<h1>Output</h1>
<p>Wrote this out to read in in the script Liver-WGCNA-Cholesterol.Qmd</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb19"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1" aria-hidden="true" tabindex="-1"></a><span class="fu">save.image</span>(<span class="at">file =</span> <span class="st">"ncd_hf_wcgna_networks.RData"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="session-information" class="level1">
<h1>Session Information</h1>
<div class="cell">
<div class="sourceCode cell-code" id="cb20"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1" aria-hidden="true" tabindex="-1"></a><span class="fu">sessionInfo</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>R version 4.4.0 (2024-04-24)
Platform: x86_64-pc-linux-gnu
Running under: Red Hat Enterprise Linux 8.8 (Ootpa)

Matrix products: default
BLAS:   /sw/pkgs/arc/stacks/gcc/13.2.0/R/4.4.0/lib64/R/lib/libRblas.so 
LAPACK: /sw/pkgs/arc/stacks/gcc/13.2.0/R/4.4.0/lib64/R/lib/libRlapack.so;  LAPACK version 3.12.0

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

time zone: America/Detroit
tzcode source: system (glibc)

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] WGCNA_1.73            fastcluster_1.2.6     dynamicTreeCut_1.63-1
[4] ggplot2_3.5.1         readr_2.1.5           dplyr_1.1.4          
[7] tidyr_1.3.1           knitr_1.49           

loaded via a namespace (and not attached):
 [1] tidyselect_1.2.1        blob_1.2.4              Biostrings_2.72.1      
 [4] fastmap_1.2.0           digest_0.6.37           rpart_4.1.23           
 [7] lifecycle_1.0.4         cluster_2.1.6           survival_3.5-8         
[10] KEGGREST_1.44.1         RSQLite_2.3.9           magrittr_2.0.3         
[13] compiler_4.4.0          rlang_1.1.4             Hmisc_5.2-2            
[16] tools_4.4.0             yaml_2.3.9              data.table_1.16.4      
[19] htmlwidgets_1.6.4       bit_4.5.0.1             withr_3.0.2            
[22] foreign_0.8-86          purrr_1.0.2             BiocGenerics_0.50.0    
[25] nnet_7.3-19             grid_4.4.0              stats4_4.4.0           
[28] preprocessCore_1.66.0   colorspace_2.1-1        GO.db_3.19.1           
[31] scales_1.3.0            iterators_1.0.14        cli_3.6.3              
[34] rmarkdown_2.29          crayon_1.5.3            generics_0.1.3         
[37] rstudioapi_0.17.1       httr_1.4.7              tzdb_0.4.0             
[40] DBI_1.2.3               cachem_1.1.0            stringr_1.5.1          
[43] zlibbioc_1.50.0         splines_4.4.0           parallel_4.4.0         
[46] impute_1.78.0           AnnotationDbi_1.66.0    XVector_0.44.0         
[49] matrixStats_1.5.0       base64enc_0.1-3         vctrs_0.6.5            
[52] Matrix_1.7-0            jsonlite_1.8.9          IRanges_2.38.1         
[55] hms_1.1.3               S4Vectors_0.42.1        bit64_4.5.2            
[58] Formula_1.2-5           htmlTable_2.4.3         foreach_1.5.2          
[61] glue_1.8.0              codetools_0.2-20        stringi_1.8.4          
[64] gtable_0.3.6            GenomeInfoDb_1.40.1     UCSC.utils_1.0.0       
[67] munsell_0.5.1           tibble_3.2.1            pillar_1.10.1          
[70] htmltools_0.5.8.1       GenomeInfoDbData_1.2.12 R6_2.5.1               
[73] doParallel_1.0.17       evaluate_1.0.1          Biobase_2.64.0         
[76] lattice_0.22-6          png_0.1-8               backports_1.5.0        
[79] memoise_2.0.1           Rcpp_1.0.13-1           gridExtra_2.3          
[82] checkmate_2.3.2         xfun_0.50               pkgconfig_2.0.3        </code></pre>
</div>
</div>
</section>

</main>
<!-- /main column -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->



</body></html>