<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.2.269">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Dave Bridges">

<title>WGCNA Analysis of DO Livers</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1.6em;
  vertical-align: middle;
}
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


<script src="Liver-WGCNA-Construction_files/libs/clipboard/clipboard.min.js"></script>
<script src="Liver-WGCNA-Construction_files/libs/quarto-html/quarto.js"></script>
<script src="Liver-WGCNA-Construction_files/libs/quarto-html/popper.min.js"></script>
<script src="Liver-WGCNA-Construction_files/libs/quarto-html/tippy.umd.min.js"></script>
<script src="Liver-WGCNA-Construction_files/libs/quarto-html/anchor.min.js"></script>
<link href="Liver-WGCNA-Construction_files/libs/quarto-html/tippy.css" rel="stylesheet">
<link href="Liver-WGCNA-Construction_files/libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="Liver-WGCNA-Construction_files/libs/bootstrap/bootstrap.min.js"></script>
<link href="Liver-WGCNA-Construction_files/libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="Liver-WGCNA-Construction_files/libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">


</head>

<body class="fullcontent">

<div id="quarto-content" class="page-columns page-rows-contents page-layout-article">

<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">WGCNA Analysis of DO Livers</h1>
<p class="subtitle lead">Construction of Modules</p>
</div>



<div class="quarto-title-meta">

    <div>
    <div class="quarto-title-meta-heading">Author</div>
    <div class="quarto-title-meta-contents">
             <p>Dave Bridges </p>
          </div>
  </div>
    
  
    
  </div>
  

</header>

<p>Followed instructions on https://horvath.genetics.ucla.edu/html/CoexpressionNetwork/Rpackages/WGCNA/Tutorials/, which was based on the method used in https://doi.org/10.1371/journal.pgen.0020130</p>
<section id="data-preparation" class="level1">
<h1>Data Preparation</h1>
<div class="cell">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(WGCNA)</span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="fu">options</span>(<span class="at">stringsAsFactors =</span> F)</span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a>expression.filename <span class="ot">&lt;-</span> <span class="st">"GSE72759_DO192_RNAseq_UpperQuartileNormalized_n21454genes_forGEOSubmission.txt"</span></span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tibble)</span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a>expression.data <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(expression.filename) <span class="sc">%&gt;%</span></span>
<span id="cb1-7"><a href="#cb1-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">gene_id=</span><span class="dv">1</span>) <span class="sc">%&gt;%</span></span>
<span id="cb1-8"><a href="#cb1-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">column_to_rownames</span>(<span class="st">'gene_id'</span>) <span class="sc">%&gt;%</span></span>
<span id="cb1-9"><a href="#cb1-9" aria-hidden="true" tabindex="-1"></a>  t <span class="sc">%&gt;%</span> as.data.frame <span class="co">#rotate with samples on the left and genes on the columns</span></span>
<span id="cb1-10"><a href="#cb1-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-11"><a href="#cb1-11" aria-hidden="true" tabindex="-1"></a>phenotype.filename <span class="ot">&lt;-</span> <span class="st">'Svenson_HFD_DO_phenotype_V12.csv'</span></span>
<span id="cb1-12"><a href="#cb1-12" aria-hidden="true" tabindex="-1"></a>phenotype.data <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(phenotype.filename) <span class="sc">%&gt;%</span> </span>
<span id="cb1-13"><a href="#cb1-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(mouse.id,chol2,sex,diet) <span class="sc">%&gt;%</span></span>
<span id="cb1-14"><a href="#cb1-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">column_to_rownames</span>(<span class="st">'mouse.id'</span>)</span>
<span id="cb1-15"><a href="#cb1-15" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb1-16"><a href="#cb1-16" aria-hidden="true" tabindex="-1"></a>phenotype.data[phenotype.data<span class="sc">==</span><span class="st">'-999999'</span>] <span class="ot">&lt;-</span> <span class="cn">NA</span></span>
<span id="cb1-17"><a href="#cb1-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-18"><a href="#cb1-18" aria-hidden="true" tabindex="-1"></a>good.genes <span class="ot">&lt;-</span> <span class="fu">goodSamplesGenes</span>(expression.data)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code> Flagging genes and samples with too many missing values...
  ..step 1</code></pre>
</div>
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="co">#check if all genes are ok, not too many missing values for genes or samples</span></span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a>good.genes<span class="sc">$</span>allOK</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] TRUE</code></pre>
</div>
</div>
</section>
<section id="network-construction" class="level1">
<h1>Network Construction</h1>
<div class="cell">
<div class="sourceCode cell-code" id="cb5"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a>nSets <span class="ot">&lt;-</span> <span class="dv">4</span> <span class="co">#four datasets, NCD and HFD, male and female</span></span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a>setLabels <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"NCD-M"</span>,<span class="st">"HFHS-M"</span>,<span class="st">"NCD-F"</span>,<span class="st">"HFHS-F"</span>)</span>
<span id="cb5-3"><a href="#cb5-3" aria-hidden="true" tabindex="-1"></a>multiExpr<span class="ot">=</span><span class="fu">vector</span>(<span class="at">mode=</span><span class="st">"list"</span>,<span class="at">length=</span>nSets) <span class="co">#create vector with two datasets</span></span>
<span id="cb5-4"><a href="#cb5-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-5"><a href="#cb5-5" aria-hidden="true" tabindex="-1"></a><span class="co">#filter for only exprerssion data where we have appropriatet phenotype data</span></span>
<span id="cb5-6"><a href="#cb5-6" aria-hidden="true" tabindex="-1"></a>chow.mice.m <span class="ot">&lt;-</span> phenotype.data <span class="sc">%&gt;%</span> <span class="fu">filter</span>(diet<span class="sc">==</span><span class="st">"chow"</span><span class="sc">&amp;</span>sex<span class="sc">==</span><span class="st">"M"</span>) <span class="sc">%&gt;%</span> rownames</span>
<span id="cb5-7"><a href="#cb5-7" aria-hidden="true" tabindex="-1"></a>hf.mice.m <span class="ot">&lt;-</span> phenotype.data <span class="sc">%&gt;%</span> <span class="fu">filter</span>(diet<span class="sc">==</span><span class="st">"hf"</span><span class="sc">&amp;</span>sex<span class="sc">==</span><span class="st">"M"</span>) <span class="sc">%&gt;%</span> rownames</span>
<span id="cb5-8"><a href="#cb5-8" aria-hidden="true" tabindex="-1"></a>chow.mice.f <span class="ot">&lt;-</span> phenotype.data <span class="sc">%&gt;%</span> <span class="fu">filter</span>(diet<span class="sc">==</span><span class="st">"chow"</span><span class="sc">&amp;</span>sex<span class="sc">==</span><span class="st">"F"</span>) <span class="sc">%&gt;%</span> rownames</span>
<span id="cb5-9"><a href="#cb5-9" aria-hidden="true" tabindex="-1"></a>hf.mice.f <span class="ot">&lt;-</span> phenotype.data <span class="sc">%&gt;%</span> <span class="fu">filter</span>(diet<span class="sc">==</span><span class="st">"hf"</span><span class="sc">&amp;</span>sex<span class="sc">==</span><span class="st">"F"</span>) <span class="sc">%&gt;%</span> rownames</span>
<span id="cb5-10"><a href="#cb5-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-11"><a href="#cb5-11" aria-hidden="true" tabindex="-1"></a>chow.expression.m <span class="ot">&lt;-</span> expression.data[chow.mice.m,] <span class="sc">%&gt;%</span> na.omit</span>
<span id="cb5-12"><a href="#cb5-12" aria-hidden="true" tabindex="-1"></a>hf.expression.m <span class="ot">&lt;-</span> expression.data[hf.mice.m,] <span class="sc">%&gt;%</span> na.omit</span>
<span id="cb5-13"><a href="#cb5-13" aria-hidden="true" tabindex="-1"></a>chow.expression.f <span class="ot">&lt;-</span> expression.data[chow.mice.f,] <span class="sc">%&gt;%</span> na.omit</span>
<span id="cb5-14"><a href="#cb5-14" aria-hidden="true" tabindex="-1"></a>hf.expression.f <span class="ot">&lt;-</span> expression.data[hf.mice.f,] <span class="sc">%&gt;%</span> na.omit</span>
<span id="cb5-15"><a href="#cb5-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-16"><a href="#cb5-16" aria-hidden="true" tabindex="-1"></a><span class="co"># enter data on expression in to multiExpr object</span></span>
<span id="cb5-17"><a href="#cb5-17" aria-hidden="true" tabindex="-1"></a>multiExpr[[<span class="dv">1</span>]]<span class="ot">=</span><span class="fu">list</span>(<span class="at">data=</span>chow.expression.m)</span>
<span id="cb5-18"><a href="#cb5-18" aria-hidden="true" tabindex="-1"></a>multiExpr[[<span class="dv">2</span>]]<span class="ot">=</span><span class="fu">list</span>(<span class="at">data=</span>hf.expression.m) </span>
<span id="cb5-19"><a href="#cb5-19" aria-hidden="true" tabindex="-1"></a>multiExpr[[<span class="dv">3</span>]]<span class="ot">=</span><span class="fu">list</span>(<span class="at">data=</span>chow.expression.f)</span>
<span id="cb5-20"><a href="#cb5-20" aria-hidden="true" tabindex="-1"></a>multiExpr[[<span class="dv">4</span>]]<span class="ot">=</span><span class="fu">list</span>(<span class="at">data=</span>hf.expression.f) </span>
<span id="cb5-21"><a href="#cb5-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-22"><a href="#cb5-22" aria-hidden="true" tabindex="-1"></a>exprSize <span class="ot">&lt;-</span> <span class="fu">checkSets</span>(multiExpr)</span>
<span id="cb5-23"><a href="#cb5-23" aria-hidden="true" tabindex="-1"></a>gsg <span class="ot">&lt;-</span> <span class="fu">goodSamplesGenesMS</span>(multiExpr)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code> Flagging genes and samples with too many missing values...
  ..step 1
  ..Excluding 78 genes from the calculation due to too many missing samples or zero variance.
  ..step 2</code></pre>
</div>
<div class="sourceCode cell-code" id="cb7"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a><span class="co"># remove genes with too many missing values or no variance</span></span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span>gsg<span class="sc">$</span>allOK)</span>
<span id="cb7-3"><a href="#cb7-3" aria-hidden="true" tabindex="-1"></a>{</span>
<span id="cb7-4"><a href="#cb7-4" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Print information about the removed genes:</span></span>
<span id="cb7-5"><a href="#cb7-5" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (<span class="fu">sum</span>(<span class="sc">!</span>gsg<span class="sc">$</span>goodGenes) <span class="sc">&gt;</span> <span class="dv">0</span>)</span>
<span id="cb7-6"><a href="#cb7-6" aria-hidden="true" tabindex="-1"></a>    <span class="fu">printFlush</span>(<span class="fu">paste</span>(<span class="st">"Removing genes:"</span>, <span class="fu">paste</span>(<span class="fu">names</span>(multiExpr[[<span class="dv">1</span>]]<span class="sc">$</span>data)[<span class="sc">!</span>gsg<span class="sc">$</span>goodGenes], </span>
<span id="cb7-7"><a href="#cb7-7" aria-hidden="true" tabindex="-1"></a>                                              <span class="at">collapse =</span> <span class="st">", "</span>)))</span>
<span id="cb7-8"><a href="#cb7-8" aria-hidden="true" tabindex="-1"></a>  <span class="cf">for</span> (set <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span>exprSize<span class="sc">$</span>nSets)</span>
<span id="cb7-9"><a href="#cb7-9" aria-hidden="true" tabindex="-1"></a>  {</span>
<span id="cb7-10"><a href="#cb7-10" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> (<span class="fu">sum</span>(<span class="sc">!</span>gsg<span class="sc">$</span>goodSamples[[set]]))</span>
<span id="cb7-11"><a href="#cb7-11" aria-hidden="true" tabindex="-1"></a>      <span class="fu">printFlush</span>(<span class="fu">paste</span>(<span class="st">"In set"</span>, setLabels[set], <span class="st">"removing samples"</span>,</span>
<span id="cb7-12"><a href="#cb7-12" aria-hidden="true" tabindex="-1"></a>                       <span class="fu">paste</span>(<span class="fu">rownames</span>(multiExpr[[set]]<span class="sc">$</span>data)[<span class="sc">!</span>gsg<span class="sc">$</span>goodSamples[[set]]], <span class="at">collapse =</span> <span class="st">", "</span>)))</span>
<span id="cb7-13"><a href="#cb7-13" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Remove the offending genes and samples</span></span>
<span id="cb7-14"><a href="#cb7-14" aria-hidden="true" tabindex="-1"></a>    multiExpr[[set]]<span class="sc">$</span>data <span class="ot">=</span> multiExpr[[set]]<span class="sc">$</span>data[gsg<span class="sc">$</span>goodSamples[[set]], gsg<span class="sc">$</span>goodGenes];</span>
<span id="cb7-15"><a href="#cb7-15" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb7-16"><a href="#cb7-16" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Update exprSize</span></span>
<span id="cb7-17"><a href="#cb7-17" aria-hidden="true" tabindex="-1"></a>  exprSize <span class="ot">=</span> <span class="fu">checkSets</span>(multiExpr)</span>
<span id="cb7-18"><a href="#cb7-18" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Removing genes: ENSMUSG00000041872, ENSMUSG00000092563, ENSMUSG00000071890, ENSMUSG00000049598, ENSMUSG00000026645, ENSMUSG00000085942, ENSMUSG00000083503, ENSMUSG00000082190, ENSMUSG00000083424, ENSMUSG00000027225, ENSMUSG00000027273, ENSMUSG00000083341, ENSMUSG00000091572, ENSMUSG00000093671, ENSMUSG00000004885, ENSMUSG00000027966, ENSMUSG00000028236, ENSMUSG00000082151, ENSMUSG00000081997, ENSMUSG00000084828, ENSMUSG00000081504, ENSMUSG00000085654, ENSMUSG00000038598, ENSMUSG00000094222, ENSMUSG00000084342, ENSMUSG00000085714, ENSMUSG00000081792, ENSMUSG00000070686, ENSMUSG00000096742, ENSMUSG00000058736, ENSMUSG00000085345, ENSMUSG00000043036, ENSMUSG00000094664, ENSMUSG00000083728, ENSMUSG00000056203, ENSMUSG00000033726, ENSMUSG00000099281, ENSMUSG00000098382, ENSMUSG00000011350, ENSMUSG00000046130, ENSMUSG00000096714, ENSMUSG00000030781, ENSMUSG00000034467, ENSMUSG00000074139, ENSMUSG00000050473, ENSMUSG00000096573, ENSMUSG00000010803, ENSMUSG00000085888, ENSMUSG00000020890, ENSMUSG00000082550, ENSMUSG00000081141, ENSMUSG00000084890, ENSMUSG00000086298, ENSMUSG00000044067, ENSMUSG00000092006, ENSMUSG00000006179, ENSMUSG00000093587, ENSMUSG00000097582, ENSMUSG00000079360, ENSMUSG00000082662, ENSMUSG00000021953, ENSMUSG00000098002, ENSMUSG00000072419, ENSMUSG00000085783, ENSMUSG00000089941, ENSMUSG00000067575, ENSMUSG00000092395, ENSMUSG00000092223, ENSMUSG00000096140, ENSMUSG00000041565, ENSMUSG00000055471, ENSMUSG00000078528, ENSMUSG00000081461, ENSMUSG00000031376, ENSMUSG00000083283, ENSMUSG00000083711, ENSMUSG00000082382, ENSMUSG00000025658</code></pre>
</div>
</div>
<section id="clustering-expression-datasets" class="level2">
<h2 class="anchored" data-anchor-id="clustering-expression-datasets">Clustering Expression Datasets</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb9"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a><span class="co">#crerate two sample trees</span></span>
<span id="cb9-2"><a href="#cb9-2" aria-hidden="true" tabindex="-1"></a>sampleTrees <span class="ot">=</span> <span class="fu">list</span>()</span>
<span id="cb9-3"><a href="#cb9-3" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (set <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span>nSets)</span>
<span id="cb9-4"><a href="#cb9-4" aria-hidden="true" tabindex="-1"></a>{</span>
<span id="cb9-5"><a href="#cb9-5" aria-hidden="true" tabindex="-1"></a>  sampleTrees[[set]] <span class="ot">=</span> <span class="fu">hclust</span>(<span class="fu">dist</span>(multiExpr[[set]]<span class="sc">$</span>data), <span class="at">method =</span> <span class="st">"average"</span>)</span>
<span id="cb9-6"><a href="#cb9-6" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb9-7"><a href="#cb9-7" aria-hidden="true" tabindex="-1"></a><span class="co">#cluster genes in each set, make a hclust tree using "average" method</span></span>
<span id="cb9-8"><a href="#cb9-8" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (set <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span>nSets)</span>
<span id="cb9-9"><a href="#cb9-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">plot</span>(sampleTrees[[set]], </span>
<span id="cb9-10"><a href="#cb9-10" aria-hidden="true" tabindex="-1"></a>       <span class="at">main =</span> <span class="fu">paste</span>(<span class="st">"Sample clustering on all genes in"</span>,</span>
<span id="cb9-11"><a href="#cb9-11" aria-hidden="true" tabindex="-1"></a>                    setLabels[set]),</span>
<span id="cb9-12"><a href="#cb9-12" aria-hidden="true" tabindex="-1"></a>       <span class="at">xlab=</span><span class="st">""</span>, </span>
<span id="cb9-13"><a href="#cb9-13" aria-hidden="true" tabindex="-1"></a>       <span class="at">sub=</span><span class="st">""</span>, </span>
<span id="cb9-14"><a href="#cb9-14" aria-hidden="true" tabindex="-1"></a>       <span class="at">cex =</span> <span class="fl">0.7</span>);</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="figures/diet-clustered-trees-1.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output-display">
<p><img src="figures/diet-clustered-trees-2.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output-display">
<p><img src="figures/diet-clustered-trees-3.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output-display">
<p><img src="figures/diet-clustered-trees-4.png" class="img-fluid" width="672"></p>
</div>
<div class="sourceCode cell-code" id="cb10"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Form a multi-set structure called Traits that will hold the clinical traits.</span></span>
<span id="cb10-2"><a href="#cb10-2" aria-hidden="true" tabindex="-1"></a>Traits <span class="ot">=</span> <span class="fu">vector</span>(<span class="at">mode=</span><span class="st">"list"</span>, <span class="at">length =</span> nSets);</span>
<span id="cb10-3"><a href="#cb10-3" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (set <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span>nSets)</span>
<span id="cb10-4"><a href="#cb10-4" aria-hidden="true" tabindex="-1"></a>{</span>
<span id="cb10-5"><a href="#cb10-5" aria-hidden="true" tabindex="-1"></a>  setSamples <span class="ot">=</span> <span class="fu">rownames</span>(multiExpr[[set]]<span class="sc">$</span>data);</span>
<span id="cb10-6"><a href="#cb10-6" aria-hidden="true" tabindex="-1"></a>  traitRows <span class="ot">=</span> <span class="fu">match</span>(setSamples, <span class="fu">rownames</span>(phenotype.data));</span>
<span id="cb10-7"><a href="#cb10-7" aria-hidden="true" tabindex="-1"></a>  Traits[[set]] <span class="ot">=</span> <span class="fu">list</span>(<span class="at">data =</span> phenotype.data[traitRows,]);</span>
<span id="cb10-8"><a href="#cb10-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rownames</span>(Traits[[set]]<span class="sc">$</span>data) <span class="ot">=</span> <span class="fu">rownames</span>(phenotype.data[traitRows,]);</span>
<span id="cb10-9"><a href="#cb10-9" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb10-10"><a href="#cb10-10" aria-hidden="true" tabindex="-1"></a><span class="fu">collectGarbage</span>();</span>
<span id="cb10-11"><a href="#cb10-11" aria-hidden="true" tabindex="-1"></a><span class="co"># Define data set dimensions</span></span>
<span id="cb10-12"><a href="#cb10-12" aria-hidden="true" tabindex="-1"></a>nGenes <span class="ot">=</span> exprSize<span class="sc">$</span>nGenes;</span>
<span id="cb10-13"><a href="#cb10-13" aria-hidden="true" tabindex="-1"></a>nSamples <span class="ot">=</span> exprSize<span class="sc">$</span>nSamples;</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>The combined dataset includes 21376 genes and 48465048 samples separated into <code>nSets</code> datasets (<code>r</code>paste(setLabels, collapse=“,”)`).</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb11"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Choose a set of soft-thresholding powers</span></span>
<span id="cb11-2"><a href="#cb11-2" aria-hidden="true" tabindex="-1"></a>powers <span class="ot">=</span> <span class="fu">seq</span>(<span class="dv">4</span>,<span class="dv">20</span>,<span class="at">by=</span><span class="dv">1</span>)</span>
<span id="cb11-3"><a href="#cb11-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Initialize a list to hold the results of scale-free analysis</span></span>
<span id="cb11-4"><a href="#cb11-4" aria-hidden="true" tabindex="-1"></a>powerTables <span class="ot">=</span> <span class="fu">vector</span>(<span class="at">mode =</span> <span class="st">"list"</span>, <span class="at">length =</span> nSets);</span>
<span id="cb11-5"><a href="#cb11-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Call the network topology analysis function for each set in turn</span></span>
<span id="cb11-6"><a href="#cb11-6" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (set <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span>nSets)</span>
<span id="cb11-7"><a href="#cb11-7" aria-hidden="true" tabindex="-1"></a>  powerTables[[set]] <span class="ot">=</span> <span class="fu">list</span>(<span class="at">data =</span> <span class="fu">pickSoftThreshold</span>(multiExpr[[set]]<span class="sc">$</span>data,</span>
<span id="cb11-8"><a href="#cb11-8" aria-hidden="true" tabindex="-1"></a>                                                     <span class="at">powerVector=</span>powers,</span>
<span id="cb11-9"><a href="#cb11-9" aria-hidden="true" tabindex="-1"></a>                                                     <span class="at">verbose =</span> <span class="dv">2</span>)[[<span class="dv">2</span>]]);</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>pickSoftThreshold: will use block size 2092.
 pickSoftThreshold: calculating connectivity for given powers...
   ..working on genes 1 through 2092 of 21376
   ..working on genes 2093 through 4184 of 21376
   ..working on genes 4185 through 6276 of 21376
   ..working on genes 6277 through 8368 of 21376
   ..working on genes 8369 through 10460 of 21376
   ..working on genes 10461 through 12552 of 21376
   ..working on genes 12553 through 14644 of 21376
   ..working on genes 14645 through 16736 of 21376
   ..working on genes 16737 through 18828 of 21376
   ..working on genes 18829 through 20920 of 21376
   ..working on genes 20921 through 21376 of 21376
   Power SFT.R.sq slope truncated.R.sq mean.k. median.k. max.k.
1      4    0.892 -2.87          0.991  65.100  45.60000  401.0
2      5    0.913 -2.59          0.998  26.700  16.00000  238.0
3      6    0.922 -2.42          0.999  12.400   6.22000  154.0
4      7    0.925 -2.28          0.997   6.420   2.62000  106.0
5      8    0.928 -2.12          0.993   3.630   1.20000   76.1
6      9    0.910 -2.01          0.963   2.210   0.58400   57.2
7     10    0.944 -1.77          0.960   1.450   0.29500   44.2
8     11    0.971 -1.59          0.962   1.010   0.15500   35.3
9     12    0.974 -1.54          0.967   0.742   0.08460   34.2
10    13    0.967 -1.48          0.960   0.572   0.04720   33.5
11    14    0.967 -1.44          0.961   0.459   0.02690   32.9
12    15    0.947 -1.40          0.936   0.382   0.01560   32.5
13    16    0.926 -1.38          0.910   0.327   0.00919   32.2
14    17    0.893 -1.36          0.865   0.287   0.00547   31.9
15    18    0.362 -1.79          0.241   0.257   0.00332   31.7
16    19    0.362 -1.74          0.240   0.233   0.00204   31.6
17    20    0.363 -1.70          0.241   0.215   0.00126   31.5
pickSoftThreshold: will use block size 2092.
 pickSoftThreshold: calculating connectivity for given powers...
   ..working on genes 1 through 2092 of 21376
   ..working on genes 2093 through 4184 of 21376
   ..working on genes 4185 through 6276 of 21376
   ..working on genes 6277 through 8368 of 21376
   ..working on genes 8369 through 10460 of 21376
   ..working on genes 10461 through 12552 of 21376
   ..working on genes 12553 through 14644 of 21376
   ..working on genes 14645 through 16736 of 21376
   ..working on genes 16737 through 18828 of 21376
   ..working on genes 18829 through 20920 of 21376
   ..working on genes 20921 through 21376 of 21376
   Power SFT.R.sq slope truncated.R.sq mean.k. median.k. max.k.
1      4    0.836 -2.34          0.970  98.800  60.50000  593.0
2      5    0.845 -2.21          0.979  44.000  22.70000  345.0
3      6    0.849 -2.12          0.983  21.700   9.45000  212.0
4      7    0.854 -2.04          0.986  11.600   4.23000  135.0
5      8    0.865 -1.95          0.993   6.690   2.03000   88.8
6      9    0.869 -1.88          0.989   4.080   1.03000   60.8
7     10    0.911 -1.70          0.989   2.620   0.54500   42.6
8     11    0.925 -1.64          0.950   1.760   0.29800   33.7
9     12    0.886 -1.76          0.880   1.240   0.16800   33.1
10    13    0.349 -2.44          0.191   0.908   0.09830   32.6
11    14    0.321 -2.88          0.143   0.690   0.05830   32.3
12    15    0.332 -2.78          0.153   0.544   0.03490   32.1
13    16    0.333 -2.66          0.153   0.442   0.02150   31.9
14    17    0.336 -2.56          0.155   0.370   0.01340   31.7
15    18    0.336 -2.46          0.154   0.317   0.00837   31.6
16    19    0.319 -2.64          0.125   0.278   0.00531   31.5
17    20    0.319 -2.56          0.126   0.249   0.00341   31.4
pickSoftThreshold: will use block size 2092.
 pickSoftThreshold: calculating connectivity for given powers...
   ..working on genes 1 through 2092 of 21376
   ..working on genes 2093 through 4184 of 21376
   ..working on genes 4185 through 6276 of 21376
   ..working on genes 6277 through 8368 of 21376
   ..working on genes 8369 through 10460 of 21376
   ..working on genes 10461 through 12552 of 21376
   ..working on genes 12553 through 14644 of 21376
   ..working on genes 14645 through 16736 of 21376
   ..working on genes 16737 through 18828 of 21376
   ..working on genes 18829 through 20920 of 21376
   ..working on genes 20921 through 21376 of 21376
   Power SFT.R.sq slope truncated.R.sq mean.k. median.k. max.k.
1      4    0.945 -2.85          0.994  55.200 39.300000  342.0
2      5    0.956 -2.60          0.994  21.500 13.300000  202.0
3      6    0.956 -2.43          0.986   9.530  4.990000  134.0
4      7    0.962 -2.27          0.989   4.720  2.050000   95.3
5      8    0.963 -2.09          0.985   2.590  0.900000   71.6
6      9    0.945 -1.93          0.959   1.550  0.421000   55.9
7     10    0.915 -1.76          0.908   1.010  0.207000   44.9
8     11    0.964 -1.52          0.962   0.714  0.105000   36.9
9     12    0.940 -1.44          0.936   0.536  0.055300   33.7
10    13    0.955 -1.36          0.960   0.425  0.029800   32.1
11    14    0.940 -1.32          0.940   0.353  0.016500   31.0
12    15    0.935 -1.27          0.935   0.303  0.009300   31.0
13    16    0.897 -1.25          0.884   0.267  0.005370   31.0
14    17    0.897 -1.24          0.881   0.241  0.003110   31.0
15    18    0.910 -1.21          0.898   0.221  0.001840   31.0
16    19    0.900 -1.20          0.885   0.205  0.001100   31.0
17    20    0.838 -1.23          0.795   0.193  0.000663   31.0
pickSoftThreshold: will use block size 2092.
 pickSoftThreshold: calculating connectivity for given powers...
   ..working on genes 1 through 2092 of 21376
   ..working on genes 2093 through 4184 of 21376
   ..working on genes 4185 through 6276 of 21376
   ..working on genes 6277 through 8368 of 21376
   ..working on genes 8369 through 10460 of 21376
   ..working on genes 10461 through 12552 of 21376
   ..working on genes 12553 through 14644 of 21376
   ..working on genes 14645 through 16736 of 21376
   ..working on genes 16737 through 18828 of 21376
   ..working on genes 18829 through 20920 of 21376
   ..working on genes 20921 through 21376 of 21376
   Power SFT.R.sq slope truncated.R.sq mean.k. median.k. max.k.
1      4    0.821 -2.31          0.950  86.800  54.20000  486.0
2      5    0.842 -2.19          0.971  38.500  19.50000  302.0
3      6    0.854 -2.09          0.982  19.200   7.80000  202.0
4      7    0.855 -1.99          0.973  10.500   3.38000  142.0
5      8    0.946 -1.72          0.981   6.270   1.57000  103.0
6      9    0.918 -1.76          0.923   4.000   0.77500   91.8
7     10    0.891 -1.78          0.871   2.700   0.39600   87.2
8     11    0.876 -1.74          0.841   1.930   0.21200   83.4
9     12    0.873 -1.68          0.837   1.440   0.11600   80.2
10    13    0.910 -1.57          0.886   1.130   0.06610   77.4
11    14    0.911 -1.50          0.896   0.912   0.03800   75.0
12    15    0.899 -1.43          0.888   0.763   0.02240   72.9
13    16    0.887 -1.38          0.880   0.656   0.01350   71.1
14    17    0.889 -1.34          0.883   0.577   0.00824   69.4
15    18    0.903 -1.29          0.899   0.518   0.00505   67.9
16    19    0.911 -1.25          0.911   0.472   0.00311   66.5
17    20    0.912 -1.22          0.915   0.436   0.00193   65.3</code></pre>
</div>
<div class="sourceCode cell-code" id="cb13"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a><span class="fu">collectGarbage</span>();</span>
<span id="cb13-2"><a href="#cb13-2" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot the results:</span></span>
<span id="cb13-3"><a href="#cb13-3" aria-hidden="true" tabindex="-1"></a>colors <span class="ot">=</span> color.scheme</span>
<span id="cb13-4"><a href="#cb13-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-5"><a href="#cb13-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Will plot these columns of the returned scale free analysis tables</span></span>
<span id="cb13-6"><a href="#cb13-6" aria-hidden="true" tabindex="-1"></a>plotCols <span class="ot">=</span> <span class="fu">c</span>(<span class="dv">2</span>,<span class="dv">5</span>,<span class="dv">6</span>,<span class="dv">7</span>)</span>
<span id="cb13-7"><a href="#cb13-7" aria-hidden="true" tabindex="-1"></a>colNames <span class="ot">=</span> <span class="fu">c</span>(<span class="st">"Scale Free Topology Model Fit"</span>, <span class="st">"Mean connectivity"</span>, <span class="st">"Median connectivity"</span>,</span>
<span id="cb13-8"><a href="#cb13-8" aria-hidden="true" tabindex="-1"></a><span class="st">"Max connectivity"</span>);</span>
<span id="cb13-9"><a href="#cb13-9" aria-hidden="true" tabindex="-1"></a><span class="co"># Get the minima and maxima of the plotted points</span></span>
<span id="cb13-10"><a href="#cb13-10" aria-hidden="true" tabindex="-1"></a>ylim <span class="ot">=</span> <span class="fu">matrix</span>(<span class="cn">NA</span>, <span class="at">nrow =</span> <span class="dv">2</span>, <span class="at">ncol =</span> <span class="dv">4</span>);</span>
<span id="cb13-11"><a href="#cb13-11" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (set <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span>nSets)</span>
<span id="cb13-12"><a href="#cb13-12" aria-hidden="true" tabindex="-1"></a>{</span>
<span id="cb13-13"><a href="#cb13-13" aria-hidden="true" tabindex="-1"></a>  <span class="cf">for</span> (col <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(plotCols))</span>
<span id="cb13-14"><a href="#cb13-14" aria-hidden="true" tabindex="-1"></a>  {</span>
<span id="cb13-15"><a href="#cb13-15" aria-hidden="true" tabindex="-1"></a>    ylim[<span class="dv">1</span>, col] <span class="ot">=</span> <span class="fu">min</span>(ylim[<span class="dv">1</span>, col], powerTables[[set]]<span class="sc">$</span>data[, plotCols[col]], <span class="at">na.rm =</span> <span class="cn">TRUE</span>);</span>
<span id="cb13-16"><a href="#cb13-16" aria-hidden="true" tabindex="-1"></a>    ylim[<span class="dv">2</span>, col] <span class="ot">=</span> <span class="fu">max</span>(ylim[<span class="dv">2</span>, col], powerTables[[set]]<span class="sc">$</span>data[, plotCols[col]], <span class="at">na.rm =</span> <span class="cn">TRUE</span>);</span>
<span id="cb13-17"><a href="#cb13-17" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb13-18"><a href="#cb13-18" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb13-19"><a href="#cb13-19" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot the quantities in the chosen columns vs. the soft thresholding power</span></span>
<span id="cb13-20"><a href="#cb13-20" aria-hidden="true" tabindex="-1"></a><span class="fu">sizeGrWindow</span>(<span class="dv">8</span>, <span class="dv">6</span>)</span>
<span id="cb13-21"><a href="#cb13-21" aria-hidden="true" tabindex="-1"></a><span class="co">#pdf(file = "Plots/scaleFreeAnalysis.pdf", wi = 8, he = 6)</span></span>
<span id="cb13-22"><a href="#cb13-22" aria-hidden="true" tabindex="-1"></a><span class="fu">par</span>(<span class="at">mfcol =</span> <span class="fu">c</span>(<span class="dv">2</span>,<span class="dv">2</span>));</span>
<span id="cb13-23"><a href="#cb13-23" aria-hidden="true" tabindex="-1"></a><span class="fu">par</span>(<span class="at">mar =</span> <span class="fu">c</span>(<span class="fl">4.2</span>, <span class="fl">4.2</span> , <span class="fl">2.2</span>, <span class="fl">0.5</span>))</span>
<span id="cb13-24"><a href="#cb13-24" aria-hidden="true" tabindex="-1"></a>cex1 <span class="ot">=</span> <span class="fl">0.7</span>;</span>
<span id="cb13-25"><a href="#cb13-25" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (col <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(plotCols)) <span class="cf">for</span> (set <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span>nSets)</span>
<span id="cb13-26"><a href="#cb13-26" aria-hidden="true" tabindex="-1"></a>{</span>
<span id="cb13-27"><a href="#cb13-27" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (set<span class="sc">==</span><span class="dv">1</span>)</span>
<span id="cb13-28"><a href="#cb13-28" aria-hidden="true" tabindex="-1"></a>  {</span>
<span id="cb13-29"><a href="#cb13-29" aria-hidden="true" tabindex="-1"></a>    <span class="fu">plot</span>(powerTables[[set]]<span class="sc">$</span>data[,<span class="dv">1</span>], <span class="sc">-</span><span class="fu">sign</span>(powerTables[[set]]<span class="sc">$</span>data[,<span class="dv">3</span>])<span class="sc">*</span>powerTables[[set]]<span class="sc">$</span>data[,<span class="dv">2</span>],</span>
<span id="cb13-30"><a href="#cb13-30" aria-hidden="true" tabindex="-1"></a>         <span class="at">xlab=</span><span class="st">"Soft Threshold (power)"</span>,<span class="at">ylab=</span>colNames[col],<span class="at">type=</span><span class="st">"n"</span>, <span class="at">ylim =</span> ylim[, col],</span>
<span id="cb13-31"><a href="#cb13-31" aria-hidden="true" tabindex="-1"></a>         <span class="at">main =</span> colNames[col]);</span>
<span id="cb13-32"><a href="#cb13-32" aria-hidden="true" tabindex="-1"></a>    <span class="fu">addGrid</span>();</span>
<span id="cb13-33"><a href="#cb13-33" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb13-34"><a href="#cb13-34" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (col<span class="sc">==</span><span class="dv">1</span>)</span>
<span id="cb13-35"><a href="#cb13-35" aria-hidden="true" tabindex="-1"></a>  {</span>
<span id="cb13-36"><a href="#cb13-36" aria-hidden="true" tabindex="-1"></a>    <span class="fu">text</span>(powerTables[[set]]<span class="sc">$</span>data[,<span class="dv">1</span>], <span class="sc">-</span><span class="fu">sign</span>(powerTables[[set]]<span class="sc">$</span>data[,<span class="dv">3</span>])<span class="sc">*</span>powerTables[[set]]<span class="sc">$</span>data[,<span class="dv">2</span>],</span>
<span id="cb13-37"><a href="#cb13-37" aria-hidden="true" tabindex="-1"></a>         <span class="at">labels=</span>powers,<span class="at">cex=</span>cex1,<span class="at">col=</span>colors[set]);</span>
<span id="cb13-38"><a href="#cb13-38" aria-hidden="true" tabindex="-1"></a>  } <span class="cf">else</span></span>
<span id="cb13-39"><a href="#cb13-39" aria-hidden="true" tabindex="-1"></a>    <span class="fu">text</span>(powerTables[[set]]<span class="sc">$</span>data[,<span class="dv">1</span>], powerTables[[set]]<span class="sc">$</span>data[,plotCols[col]],</span>
<span id="cb13-40"><a href="#cb13-40" aria-hidden="true" tabindex="-1"></a>         <span class="at">labels=</span>powers,<span class="at">cex=</span>cex1,<span class="at">col=</span>colors[set]);</span>
<span id="cb13-41"><a href="#cb13-41" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (col<span class="sc">==</span><span class="dv">1</span>)</span>
<span id="cb13-42"><a href="#cb13-42" aria-hidden="true" tabindex="-1"></a>  {</span>
<span id="cb13-43"><a href="#cb13-43" aria-hidden="true" tabindex="-1"></a>    <span class="fu">legend</span>(<span class="st">"bottomright"</span>, <span class="at">legend =</span> setLabels, <span class="at">col =</span> colors, <span class="at">pch =</span> <span class="dv">20</span>) ;</span>
<span id="cb13-44"><a href="#cb13-44" aria-hidden="true" tabindex="-1"></a>  } <span class="cf">else</span></span>
<span id="cb13-45"><a href="#cb13-45" aria-hidden="true" tabindex="-1"></a>    <span class="fu">legend</span>(<span class="st">"topright"</span>, <span class="at">legend =</span> setLabels, <span class="at">col =</span> colors, <span class="at">pch =</span> <span class="dv">20</span>) ;</span>
<span id="cb13-46"><a href="#cb13-46" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb13-47"><a href="#cb13-47" aria-hidden="true" tabindex="-1"></a><span class="co">#dev.off();</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb14"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a>network.power <span class="ot">&lt;-</span><span class="dv">6</span></span>
<span id="cb14-2"><a href="#cb14-2" aria-hidden="true" tabindex="-1"></a>net <span class="ot">=</span> <span class="fu">blockwiseConsensusModules</span>(</span>
<span id="cb14-3"><a href="#cb14-3" aria-hidden="true" tabindex="-1"></a>        multiExpr, </span>
<span id="cb14-4"><a href="#cb14-4" aria-hidden="true" tabindex="-1"></a>        <span class="at">power =</span> network.power, </span>
<span id="cb14-5"><a href="#cb14-5" aria-hidden="true" tabindex="-1"></a>        <span class="at">minModuleSize =</span> <span class="dv">20</span>, <span class="co">#default is 20, tutorial said 30</span></span>
<span id="cb14-6"><a href="#cb14-6" aria-hidden="true" tabindex="-1"></a>        <span class="at">deepSplit =</span> <span class="dv">2</span>, <span class="co">#default</span></span>
<span id="cb14-7"><a href="#cb14-7" aria-hidden="true" tabindex="-1"></a>        <span class="at">maxBlockSize =</span> <span class="fu">nrow</span>(chow.expression.f),</span>
<span id="cb14-8"><a href="#cb14-8" aria-hidden="true" tabindex="-1"></a>        <span class="at">pamRespectsDendro =</span> <span class="cn">FALSE</span>, </span>
<span id="cb14-9"><a href="#cb14-9" aria-hidden="true" tabindex="-1"></a>        <span class="at">mergeCutHeight =</span> <span class="fl">0.25</span>,</span>
<span id="cb14-10"><a href="#cb14-10" aria-hidden="true" tabindex="-1"></a>        <span class="at">numericLabels =</span> <span class="cn">TRUE</span>,</span>
<span id="cb14-11"><a href="#cb14-11" aria-hidden="true" tabindex="-1"></a>        <span class="at">minKMEtoStay =</span> <span class="fl">0.1</span>, <span class="co">#default is 0.2, tutorial said 0.1</span></span>
<span id="cb14-12"><a href="#cb14-12" aria-hidden="true" tabindex="-1"></a>        <span class="at">saveTOMs =</span> <span class="cn">TRUE</span>, </span>
<span id="cb14-13"><a href="#cb14-13" aria-hidden="true" tabindex="-1"></a>        <span class="at">verbose =</span> <span class="dv">1</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code> Calculating consensus modules and module eigengenes block-wise from all genes
 Calculating topological overlaps block-wise from all genes</code></pre>
</div>
</div>
<p>Constructed modules with 6 soft-thresholding power, based on the visualization of the .</p>
</section>
<section id="plotting-samples-clustered-by-consensus-modules" class="level2">
<h2 class="anchored" data-anchor-id="plotting-samples-clustered-by-consensus-modules">Plotting Samples Clustered by Consensus Modules</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb16"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" aria-hidden="true" tabindex="-1"></a>consMEs <span class="ot">=</span> net<span class="sc">$</span>multiMEs</span>
<span id="cb16-2"><a href="#cb16-2" aria-hidden="true" tabindex="-1"></a>moduleLabels <span class="ot">=</span> net<span class="sc">$</span>colors</span>
<span id="cb16-3"><a href="#cb16-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Convert the numeric labels to color labels</span></span>
<span id="cb16-4"><a href="#cb16-4" aria-hidden="true" tabindex="-1"></a>moduleColors <span class="ot">=</span> <span class="fu">labels2colors</span>(moduleLabels)</span>
<span id="cb16-5"><a href="#cb16-5" aria-hidden="true" tabindex="-1"></a>consTree <span class="ot">=</span> net<span class="sc">$</span>dendrograms[[<span class="dv">1</span>]]</span>
<span id="cb16-6"><a href="#cb16-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-7"><a href="#cb16-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Convert labels to colors for plotting</span></span>
<span id="cb16-8"><a href="#cb16-8" aria-hidden="true" tabindex="-1"></a>mergedColors <span class="ot">=</span> <span class="fu">labels2colors</span>(net<span class="sc">$</span>colors)</span>
<span id="cb16-9"><a href="#cb16-9" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot the dendrogram and the module colors underneath</span></span>
<span id="cb16-10"><a href="#cb16-10" aria-hidden="true" tabindex="-1"></a><span class="fu">plotDendroAndColors</span>(net<span class="sc">$</span>dendrograms[[<span class="dv">1</span>]], mergedColors[net<span class="sc">$</span>blockGenes[[<span class="dv">1</span>]]],</span>
<span id="cb16-11"><a href="#cb16-11" aria-hidden="true" tabindex="-1"></a>                    <span class="st">"Module colors"</span>,</span>
<span id="cb16-12"><a href="#cb16-12" aria-hidden="true" tabindex="-1"></a>                    <span class="at">dendroLabels =</span> <span class="cn">FALSE</span>, <span class="at">hang =</span> <span class="fl">0.03</span>,</span>
<span id="cb16-13"><a href="#cb16-13" aria-hidden="true" tabindex="-1"></a>                    <span class="at">addGuide =</span> <span class="cn">TRUE</span>, <span class="at">guideHang =</span> <span class="fl">0.05</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="figures/wgcna-dendrograms-by-modules-1.png" class="img-fluid" width="672"></p>
</div>
<div class="sourceCode cell-code" id="cb17"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Convert labels to colors for plotting</span></span>
<span id="cb17-2"><a href="#cb17-2" aria-hidden="true" tabindex="-1"></a>mergedColors <span class="ot">=</span> <span class="fu">labels2colors</span>(net<span class="sc">$</span>colors)</span>
<span id="cb17-3"><a href="#cb17-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot the dendrogram and the module colors underneath</span></span>
<span id="cb17-4"><a href="#cb17-4" aria-hidden="true" tabindex="-1"></a><span class="fu">plotDendroAndColors</span>(net<span class="sc">$</span>dendrograms[[<span class="dv">2</span>]], mergedColors[net<span class="sc">$</span>blockGenes[[<span class="dv">1</span>]]],</span>
<span id="cb17-5"><a href="#cb17-5" aria-hidden="true" tabindex="-1"></a>                    <span class="st">"Module colors"</span>,</span>
<span id="cb17-6"><a href="#cb17-6" aria-hidden="true" tabindex="-1"></a>                    <span class="at">dendroLabels =</span> <span class="cn">FALSE</span>, <span class="at">hang =</span> <span class="fl">0.03</span>,</span>
<span id="cb17-7"><a href="#cb17-7" aria-hidden="true" tabindex="-1"></a>                    <span class="at">addGuide =</span> <span class="cn">TRUE</span>, <span class="at">guideHang =</span> <span class="fl">0.05</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="figures/wgcna-dendrograms-by-modules-2.png" class="img-fluid" width="672"></p>
</div>
<div class="sourceCode cell-code" id="cb18"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1" aria-hidden="true" tabindex="-1"></a><span class="co"># plotDendroAndColors(consTree, moduleColors,</span></span>
<span id="cb18-2"><a href="#cb18-2" aria-hidden="true" tabindex="-1"></a><span class="co">#                     "Module colors",</span></span>
<span id="cb18-3"><a href="#cb18-3" aria-hidden="true" tabindex="-1"></a><span class="co">#                     dendroLabels = FALSE, hang = 0.03,</span></span>
<span id="cb18-4"><a href="#cb18-4" aria-hidden="true" tabindex="-1"></a><span class="co">#                     addGuide = TRUE, guideHang = 0.05,</span></span>
<span id="cb18-5"><a href="#cb18-5" aria-hidden="true" tabindex="-1"></a><span class="co">#                     main = "Consensus gene dendrogram and module colors")</span></span>
<span id="cb18-6"><a href="#cb18-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-7"><a href="#cb18-7" aria-hidden="true" tabindex="-1"></a>moduleLabels <span class="ot">=</span> net<span class="sc">$</span>colors</span>
<span id="cb18-8"><a href="#cb18-8" aria-hidden="true" tabindex="-1"></a>moduleColors <span class="ot">=</span> <span class="fu">labels2colors</span>(net<span class="sc">$</span>colors)</span>
<span id="cb18-9"><a href="#cb18-9" aria-hidden="true" tabindex="-1"></a>MEs <span class="ot">=</span> net<span class="sc">$</span>MEs;</span>
<span id="cb18-10"><a href="#cb18-10" aria-hidden="true" tabindex="-1"></a>geneTree <span class="ot">=</span> net<span class="sc">$</span>dendrograms[[<span class="dv">1</span>]];</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="output" class="level1">
<h1>Output</h1>
<p>Wrote this out to read in in the script Liver-WGCNA-Cholesterol.Qmd</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb19"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1" aria-hidden="true" tabindex="-1"></a><span class="fu">save.image</span>(<span class="at">file =</span> <span class="st">"ncd_hf_wcgna_networks.RData"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="session-information" class="level1">
<h1>Session Information</h1>
<div class="cell">
<div class="sourceCode cell-code" id="cb20"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1" aria-hidden="true" tabindex="-1"></a><span class="fu">sessionInfo</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>R version 4.2.2 (2022-10-31)
Platform: x86_64-apple-darwin17.0 (64-bit)
Running under: macOS Big Sur ... 10.16

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.2/Resources/lib/libRblas.0.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/4.2/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] tibble_3.2.1          WGCNA_1.72-1          fastcluster_1.2.3    
[4] dynamicTreeCut_1.63-1 ggplot2_3.4.2         readr_2.1.4          
[7] dplyr_1.1.2           tidyr_1.3.0           knitr_1.43           

loaded via a namespace (and not attached):
 [1] Biobase_2.58.0         httr_1.4.6             vroom_1.6.3           
 [4] bit64_4.0.5            jsonlite_1.8.5         splines_4.2.2         
 [7] foreach_1.5.2          Formula_1.2-5          stats4_4.2.2          
[10] blob_1.2.4             GenomeInfoDbData_1.2.9 impute_1.70.0         
[13] yaml_2.3.7             backports_1.4.1        pillar_1.9.0          
[16] RSQLite_2.3.1          lattice_0.21-8         glue_1.6.2            
[19] digest_0.6.31          checkmate_2.2.0        XVector_0.38.0        
[22] colorspace_2.1-0       preprocessCore_1.58.0  htmltools_0.5.5       
[25] Matrix_1.5-4.1         pkgconfig_2.0.3        zlibbioc_1.44.0       
[28] purrr_1.0.1            GO.db_3.16.0           scales_1.2.1          
[31] tzdb_0.4.0             htmlTable_2.4.1        KEGGREST_1.38.0       
[34] generics_0.1.3         IRanges_2.32.0         cachem_1.0.8          
[37] withr_2.5.0            nnet_7.3-19            BiocGenerics_0.44.0   
[40] cli_3.6.1              survival_3.5-5         magrittr_2.0.3        
[43] crayon_1.5.2           memoise_2.0.1          evaluate_0.21         
[46] fansi_1.0.4            doParallel_1.0.17      foreign_0.8-84        
[49] data.table_1.14.8      tools_4.2.2            hms_1.1.3             
[52] lifecycle_1.0.3        matrixStats_1.0.0      stringr_1.5.0         
[55] S4Vectors_0.36.2       munsell_0.5.0          cluster_2.1.4         
[58] AnnotationDbi_1.60.2   Biostrings_2.66.0      compiler_4.2.2        
[61] GenomeInfoDb_1.34.9    rlang_1.1.1            grid_4.2.2            
[64] RCurl_1.98-1.12        iterators_1.0.14       rstudioapi_0.14       
[67] htmlwidgets_1.6.2      bitops_1.0-7           base64enc_0.1-3       
[70] rmarkdown_2.22         gtable_0.3.3           codetools_0.2-19      
[73] DBI_1.1.3              R6_2.5.1               gridExtra_2.3         
[76] fastmap_1.1.1          bit_4.0.5              utf8_1.2.3            
[79] Hmisc_5.1-0            stringi_1.7.12         parallel_4.2.2        
[82] Rcpp_1.0.10            vctrs_0.6.2            rpart_4.1.19          
[85] png_0.1-8              tidyselect_1.2.0       xfun_0.39             </code></pre>
</div>
</div>
</section>

</main>
<!-- /main column -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    target: function(trigger) {
      return trigger.previousElementSibling;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->



</body></html>