<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.3.450">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Dave Bridges">

<title>Creating Bimbam file from HFD-DO Viewer Dataset</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="Bimbam-generation_files/libs/clipboard/clipboard.min.js"></script>
<script src="Bimbam-generation_files/libs/quarto-html/quarto.js"></script>
<script src="Bimbam-generation_files/libs/quarto-html/popper.min.js"></script>
<script src="Bimbam-generation_files/libs/quarto-html/tippy.umd.min.js"></script>
<script src="Bimbam-generation_files/libs/quarto-html/anchor.min.js"></script>
<link href="Bimbam-generation_files/libs/quarto-html/tippy.css" rel="stylesheet">
<link href="Bimbam-generation_files/libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="Bimbam-generation_files/libs/bootstrap/bootstrap.min.js"></script>
<link href="Bimbam-generation_files/libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="Bimbam-generation_files/libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">


</head>

<body class="fullcontent">

<div id="quarto-content" class="page-columns page-rows-contents page-layout-article">

<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">Creating Bimbam file from HFD-DO Viewer Dataset</h1>
</div>



<div class="quarto-title-meta">

    <div>
    <div class="quarto-title-meta-heading">Author</div>
    <div class="quarto-title-meta-contents">
             <p>Dave Bridges </p>
          </div>
  </div>
    
  
    
  </div>
  

</header>

<section id="load-dataset" class="level1">
<h1>Load Dataset</h1>
<ul>
<li>Datafile originally at</li>
</ul>
<div class="cell">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a>filename <span class="ot">&lt;-</span> <span class="st">'core.Svenson_DO_HFD.v12.Rdata'</span></span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="fu">load</span>(filename)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="bimbam-file-format" class="level1">
<h1>Bimbam file format</h1>
<p>looks like this (from https://www.xzlab.org/software/GEMMAmanual.pdf)</p>
<pre><code>SNP,Minor,Major,ID1,ID2,ID3
rs1,A,T,0.02,0.80,1.50 
rs2,G,C,0.98,0.04,1.00</code></pre>
<div class="cell">
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyr)</span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(dplyr)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'dplyr'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:stats':

    filter, lag</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:base':

    intersect, setdiff, setequal, union</code></pre>
</div>
<div class="sourceCode cell-code" id="cb7"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tibble)</span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(stringr)</span>
<span id="cb7-3"><a href="#cb7-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-4"><a href="#cb7-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-5"><a href="#cb7-5" aria-hidden="true" tabindex="-1"></a>df<span class="fl">.1</span> <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(genoprobs<span class="sc">$</span><span class="st">`</span><span class="at">1</span><span class="st">`</span>, <span class="at">stringsAsFactors =</span> <span class="cn">FALSE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb7-6"><a href="#cb7-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="at">cols =</span> <span class="fu">everything</span>(<span class="sc">-</span><span class="dv">1</span>),  <span class="co"># Replace "Sample" with the actual name of your sample identifier</span></span>
<span id="cb7-7"><a href="#cb7-7" aria-hidden="true" tabindex="-1"></a>               <span class="at">names_to =</span> <span class="st">"Sample"</span>,</span>
<span id="cb7-8"><a href="#cb7-8" aria-hidden="true" tabindex="-1"></a>               <span class="at">values_to =</span> <span class="st">"Value"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb7-9"><a href="#cb7-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Sample=</span><span class="fu">dimnames</span>(K<span class="sc">$</span>X)[[<span class="dv">1</span>]]) <span class="sc">%&gt;%</span> <span class="co">#added sample names which were lost in the previous step</span></span>
<span id="cb7-10"><a href="#cb7-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="sc">-</span>Sample,</span>
<span id="cb7-11"><a href="#cb7-11" aria-hidden="true" tabindex="-1"></a>               <span class="at">names_to=</span><span class="st">"Genotype.ID"</span>,</span>
<span id="cb7-12"><a href="#cb7-12" aria-hidden="true" tabindex="-1"></a>               <span class="at">values_to=</span><span class="st">"Likelihood"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb7-13"><a href="#cb7-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">separate</span>(Genotype.ID, <span class="at">sep=</span><span class="dv">2</span>, <span class="at">into=</span><span class="fu">c</span>(<span class="st">"Allele"</span>,<span class="st">"Locus"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb7-14"><a href="#cb7-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Allele=</span><span class="fu">str_trunc</span>(Allele,<span class="dv">1</span>,<span class="at">side=</span><span class="st">"right"</span>, <span class="at">ellipsis=</span><span class="st">""</span>))</span>
<span id="cb7-15"><a href="#cb7-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-16"><a href="#cb7-16" aria-hidden="true" tabindex="-1"></a>df<span class="fl">.2</span> <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(genoprobs<span class="sc">$</span><span class="st">`</span><span class="at">2</span><span class="st">`</span>, <span class="at">stringsAsFactors =</span> <span class="cn">FALSE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb7-17"><a href="#cb7-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="at">cols =</span> <span class="fu">everything</span>(<span class="sc">-</span><span class="dv">1</span>),  <span class="co"># Replace "Sample" with the actual name of your sample identifier</span></span>
<span id="cb7-18"><a href="#cb7-18" aria-hidden="true" tabindex="-1"></a>               <span class="at">names_to =</span> <span class="st">"Sample"</span>,</span>
<span id="cb7-19"><a href="#cb7-19" aria-hidden="true" tabindex="-1"></a>               <span class="at">values_to =</span> <span class="st">"Value"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb7-20"><a href="#cb7-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Sample=</span><span class="fu">dimnames</span>(K<span class="sc">$</span>X)[[<span class="dv">1</span>]]) <span class="sc">%&gt;%</span> <span class="co">#added sample names which were lost in the previous step</span></span>
<span id="cb7-21"><a href="#cb7-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="sc">-</span>Sample,</span>
<span id="cb7-22"><a href="#cb7-22" aria-hidden="true" tabindex="-1"></a>               <span class="at">names_to=</span><span class="st">"Genotype.ID"</span>,</span>
<span id="cb7-23"><a href="#cb7-23" aria-hidden="true" tabindex="-1"></a>               <span class="at">values_to=</span><span class="st">"Likelihood"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb7-24"><a href="#cb7-24" aria-hidden="true" tabindex="-1"></a>  <span class="fu">separate</span>(Genotype.ID, <span class="at">sep=</span><span class="dv">2</span>, <span class="at">into=</span><span class="fu">c</span>(<span class="st">"Allele"</span>,<span class="st">"Locus"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb7-25"><a href="#cb7-25" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Allele=</span><span class="fu">str_trunc</span>(Allele,<span class="dv">1</span>,<span class="at">side=</span><span class="st">"right"</span>, <span class="at">ellipsis=</span><span class="st">""</span>))</span>
<span id="cb7-26"><a href="#cb7-26" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-27"><a href="#cb7-27" aria-hidden="true" tabindex="-1"></a>df<span class="fl">.3</span> <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(genoprobs<span class="sc">$</span><span class="st">`</span><span class="at">3</span><span class="st">`</span>, <span class="at">stringsAsFactors =</span> <span class="cn">FALSE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb7-28"><a href="#cb7-28" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="at">cols =</span> <span class="fu">everything</span>(<span class="sc">-</span><span class="dv">1</span>),  <span class="co"># Replace "Sample" with the actual name of your sample identifier</span></span>
<span id="cb7-29"><a href="#cb7-29" aria-hidden="true" tabindex="-1"></a>               <span class="at">names_to =</span> <span class="st">"Sample"</span>,</span>
<span id="cb7-30"><a href="#cb7-30" aria-hidden="true" tabindex="-1"></a>               <span class="at">values_to =</span> <span class="st">"Value"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb7-31"><a href="#cb7-31" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Sample=</span><span class="fu">dimnames</span>(K<span class="sc">$</span>X)[[<span class="dv">1</span>]]) <span class="sc">%&gt;%</span> <span class="co">#added sample names which were lost in the previous step</span></span>
<span id="cb7-32"><a href="#cb7-32" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="sc">-</span>Sample,</span>
<span id="cb7-33"><a href="#cb7-33" aria-hidden="true" tabindex="-1"></a>               <span class="at">names_to=</span><span class="st">"Genotype.ID"</span>,</span>
<span id="cb7-34"><a href="#cb7-34" aria-hidden="true" tabindex="-1"></a>               <span class="at">values_to=</span><span class="st">"Likelihood"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb7-35"><a href="#cb7-35" aria-hidden="true" tabindex="-1"></a>  <span class="fu">separate</span>(Genotype.ID, <span class="at">sep=</span><span class="dv">2</span>, <span class="at">into=</span><span class="fu">c</span>(<span class="st">"Allele"</span>,<span class="st">"Locus"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb7-36"><a href="#cb7-36" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Allele=</span><span class="fu">str_trunc</span>(Allele,<span class="dv">1</span>,<span class="at">side=</span><span class="st">"right"</span>, <span class="at">ellipsis=</span><span class="st">""</span>))</span>
<span id="cb7-37"><a href="#cb7-37" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-38"><a href="#cb7-38" aria-hidden="true" tabindex="-1"></a>df<span class="fl">.4</span> <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(genoprobs<span class="sc">$</span><span class="st">`</span><span class="at">4</span><span class="st">`</span>, <span class="at">stringsAsFactors =</span> <span class="cn">FALSE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb7-39"><a href="#cb7-39" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="at">cols =</span> <span class="fu">everything</span>(<span class="sc">-</span><span class="dv">1</span>),  <span class="co"># Replace "Sample" with the actual name of your sample identifier</span></span>
<span id="cb7-40"><a href="#cb7-40" aria-hidden="true" tabindex="-1"></a>               <span class="at">names_to =</span> <span class="st">"Sample"</span>,</span>
<span id="cb7-41"><a href="#cb7-41" aria-hidden="true" tabindex="-1"></a>               <span class="at">values_to =</span> <span class="st">"Value"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb7-42"><a href="#cb7-42" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Sample=</span><span class="fu">dimnames</span>(K<span class="sc">$</span>X)[[<span class="dv">1</span>]]) <span class="sc">%&gt;%</span> <span class="co">#added sample names which were lost in the previous step</span></span>
<span id="cb7-43"><a href="#cb7-43" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="sc">-</span>Sample,</span>
<span id="cb7-44"><a href="#cb7-44" aria-hidden="true" tabindex="-1"></a>               <span class="at">names_to=</span><span class="st">"Genotype.ID"</span>,</span>
<span id="cb7-45"><a href="#cb7-45" aria-hidden="true" tabindex="-1"></a>               <span class="at">values_to=</span><span class="st">"Likelihood"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb7-46"><a href="#cb7-46" aria-hidden="true" tabindex="-1"></a>  <span class="fu">separate</span>(Genotype.ID, <span class="at">sep=</span><span class="dv">2</span>, <span class="at">into=</span><span class="fu">c</span>(<span class="st">"Allele"</span>,<span class="st">"Locus"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb7-47"><a href="#cb7-47" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Allele=</span><span class="fu">str_trunc</span>(Allele,<span class="dv">1</span>,<span class="at">side=</span><span class="st">"right"</span>, <span class="at">ellipsis=</span><span class="st">""</span>))</span>
<span id="cb7-48"><a href="#cb7-48" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-49"><a href="#cb7-49" aria-hidden="true" tabindex="-1"></a>df<span class="fl">.5</span> <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(genoprobs<span class="sc">$</span><span class="st">`</span><span class="at">5</span><span class="st">`</span>, <span class="at">stringsAsFactors =</span> <span class="cn">FALSE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb7-50"><a href="#cb7-50" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="at">cols =</span> <span class="fu">everything</span>(<span class="sc">-</span><span class="dv">1</span>),  <span class="co"># Replace "Sample" with the actual name of your sample identifier</span></span>
<span id="cb7-51"><a href="#cb7-51" aria-hidden="true" tabindex="-1"></a>               <span class="at">names_to =</span> <span class="st">"Sample"</span>,</span>
<span id="cb7-52"><a href="#cb7-52" aria-hidden="true" tabindex="-1"></a>               <span class="at">values_to =</span> <span class="st">"Value"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb7-53"><a href="#cb7-53" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Sample=</span><span class="fu">dimnames</span>(K<span class="sc">$</span>X)[[<span class="dv">1</span>]]) <span class="sc">%&gt;%</span> <span class="co">#added sample names which were lost in the previous step</span></span>
<span id="cb7-54"><a href="#cb7-54" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="sc">-</span>Sample,</span>
<span id="cb7-55"><a href="#cb7-55" aria-hidden="true" tabindex="-1"></a>               <span class="at">names_to=</span><span class="st">"Genotype.ID"</span>,</span>
<span id="cb7-56"><a href="#cb7-56" aria-hidden="true" tabindex="-1"></a>               <span class="at">values_to=</span><span class="st">"Likelihood"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb7-57"><a href="#cb7-57" aria-hidden="true" tabindex="-1"></a>  <span class="fu">separate</span>(Genotype.ID, <span class="at">sep=</span><span class="dv">2</span>, <span class="at">into=</span><span class="fu">c</span>(<span class="st">"Allele"</span>,<span class="st">"Locus"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb7-58"><a href="#cb7-58" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Allele=</span><span class="fu">str_trunc</span>(Allele,<span class="dv">1</span>,<span class="at">side=</span><span class="st">"right"</span>, <span class="at">ellipsis=</span><span class="st">""</span>))</span>
<span id="cb7-59"><a href="#cb7-59" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-60"><a href="#cb7-60" aria-hidden="true" tabindex="-1"></a>df<span class="fl">.6</span> <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(genoprobs<span class="sc">$</span><span class="st">`</span><span class="at">6</span><span class="st">`</span>, <span class="at">stringsAsFactors =</span> <span class="cn">FALSE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb7-61"><a href="#cb7-61" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="at">cols =</span> <span class="fu">everything</span>(<span class="sc">-</span><span class="dv">1</span>),  <span class="co"># Replace "Sample" with the actual name of your sample identifier</span></span>
<span id="cb7-62"><a href="#cb7-62" aria-hidden="true" tabindex="-1"></a>               <span class="at">names_to =</span> <span class="st">"Sample"</span>,</span>
<span id="cb7-63"><a href="#cb7-63" aria-hidden="true" tabindex="-1"></a>               <span class="at">values_to =</span> <span class="st">"Value"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb7-64"><a href="#cb7-64" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Sample=</span><span class="fu">dimnames</span>(K<span class="sc">$</span>X)[[<span class="dv">1</span>]]) <span class="sc">%&gt;%</span> <span class="co">#added sample names which were lost in the previous step</span></span>
<span id="cb7-65"><a href="#cb7-65" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="sc">-</span>Sample,</span>
<span id="cb7-66"><a href="#cb7-66" aria-hidden="true" tabindex="-1"></a>               <span class="at">names_to=</span><span class="st">"Genotype.ID"</span>,</span>
<span id="cb7-67"><a href="#cb7-67" aria-hidden="true" tabindex="-1"></a>               <span class="at">values_to=</span><span class="st">"Likelihood"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb7-68"><a href="#cb7-68" aria-hidden="true" tabindex="-1"></a>  <span class="fu">separate</span>(Genotype.ID, <span class="at">sep=</span><span class="dv">2</span>, <span class="at">into=</span><span class="fu">c</span>(<span class="st">"Allele"</span>,<span class="st">"Locus"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb7-69"><a href="#cb7-69" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Allele=</span><span class="fu">str_trunc</span>(Allele,<span class="dv">1</span>,<span class="at">side=</span><span class="st">"right"</span>, <span class="at">ellipsis=</span><span class="st">""</span>))</span>
<span id="cb7-70"><a href="#cb7-70" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-71"><a href="#cb7-71" aria-hidden="true" tabindex="-1"></a>df<span class="fl">.7</span> <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(genoprobs<span class="sc">$</span><span class="st">`</span><span class="at">7</span><span class="st">`</span>, <span class="at">stringsAsFactors =</span> <span class="cn">FALSE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb7-72"><a href="#cb7-72" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="at">cols =</span> <span class="fu">everything</span>(<span class="sc">-</span><span class="dv">1</span>),  <span class="co"># Replace "Sample" with the actual name of your sample identifier</span></span>
<span id="cb7-73"><a href="#cb7-73" aria-hidden="true" tabindex="-1"></a>               <span class="at">names_to =</span> <span class="st">"Sample"</span>,</span>
<span id="cb7-74"><a href="#cb7-74" aria-hidden="true" tabindex="-1"></a>               <span class="at">values_to =</span> <span class="st">"Value"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb7-75"><a href="#cb7-75" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Sample=</span><span class="fu">dimnames</span>(K<span class="sc">$</span>X)[[<span class="dv">1</span>]]) <span class="sc">%&gt;%</span> <span class="co">#added sample names which were lost in the previous step</span></span>
<span id="cb7-76"><a href="#cb7-76" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="sc">-</span>Sample,</span>
<span id="cb7-77"><a href="#cb7-77" aria-hidden="true" tabindex="-1"></a>               <span class="at">names_to=</span><span class="st">"Genotype.ID"</span>,</span>
<span id="cb7-78"><a href="#cb7-78" aria-hidden="true" tabindex="-1"></a>               <span class="at">values_to=</span><span class="st">"Likelihood"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb7-79"><a href="#cb7-79" aria-hidden="true" tabindex="-1"></a>  <span class="fu">separate</span>(Genotype.ID, <span class="at">sep=</span><span class="dv">2</span>, <span class="at">into=</span><span class="fu">c</span>(<span class="st">"Allele"</span>,<span class="st">"Locus"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb7-80"><a href="#cb7-80" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Allele=</span><span class="fu">str_trunc</span>(Allele,<span class="dv">1</span>,<span class="at">side=</span><span class="st">"right"</span>, <span class="at">ellipsis=</span><span class="st">""</span>))</span>
<span id="cb7-81"><a href="#cb7-81" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-82"><a href="#cb7-82" aria-hidden="true" tabindex="-1"></a>df<span class="fl">.8</span> <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(genoprobs<span class="sc">$</span><span class="st">`</span><span class="at">8</span><span class="st">`</span>, <span class="at">stringsAsFactors =</span> <span class="cn">FALSE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb7-83"><a href="#cb7-83" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="at">cols =</span> <span class="fu">everything</span>(<span class="sc">-</span><span class="dv">1</span>),  <span class="co"># Replace "Sample" with the actual name of your sample identifier</span></span>
<span id="cb7-84"><a href="#cb7-84" aria-hidden="true" tabindex="-1"></a>               <span class="at">names_to =</span> <span class="st">"Sample"</span>,</span>
<span id="cb7-85"><a href="#cb7-85" aria-hidden="true" tabindex="-1"></a>               <span class="at">values_to =</span> <span class="st">"Value"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb7-86"><a href="#cb7-86" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Sample=</span><span class="fu">dimnames</span>(K<span class="sc">$</span>X)[[<span class="dv">1</span>]]) <span class="sc">%&gt;%</span> <span class="co">#added sample names which were lost in the previous step</span></span>
<span id="cb7-87"><a href="#cb7-87" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="sc">-</span>Sample,</span>
<span id="cb7-88"><a href="#cb7-88" aria-hidden="true" tabindex="-1"></a>               <span class="at">names_to=</span><span class="st">"Genotype.ID"</span>,</span>
<span id="cb7-89"><a href="#cb7-89" aria-hidden="true" tabindex="-1"></a>               <span class="at">values_to=</span><span class="st">"Likelihood"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb7-90"><a href="#cb7-90" aria-hidden="true" tabindex="-1"></a>  <span class="fu">separate</span>(Genotype.ID, <span class="at">sep=</span><span class="dv">2</span>, <span class="at">into=</span><span class="fu">c</span>(<span class="st">"Allele"</span>,<span class="st">"Locus"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb7-91"><a href="#cb7-91" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Allele=</span><span class="fu">str_trunc</span>(Allele,<span class="dv">1</span>,<span class="at">side=</span><span class="st">"right"</span>, <span class="at">ellipsis=</span><span class="st">""</span>))</span>
<span id="cb7-92"><a href="#cb7-92" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-93"><a href="#cb7-93" aria-hidden="true" tabindex="-1"></a>df<span class="fl">.9</span> <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(genoprobs<span class="sc">$</span><span class="st">`</span><span class="at">9</span><span class="st">`</span>, <span class="at">stringsAsFactors =</span> <span class="cn">FALSE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb7-94"><a href="#cb7-94" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="at">cols =</span> <span class="fu">everything</span>(<span class="sc">-</span><span class="dv">1</span>),  <span class="co"># Replace "Sample" with the actual name of your sample identifier</span></span>
<span id="cb7-95"><a href="#cb7-95" aria-hidden="true" tabindex="-1"></a>               <span class="at">names_to =</span> <span class="st">"Sample"</span>,</span>
<span id="cb7-96"><a href="#cb7-96" aria-hidden="true" tabindex="-1"></a>               <span class="at">values_to =</span> <span class="st">"Value"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb7-97"><a href="#cb7-97" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Sample=</span><span class="fu">dimnames</span>(K<span class="sc">$</span>X)[[<span class="dv">1</span>]]) <span class="sc">%&gt;%</span> <span class="co">#added sample names which were lost in the previous step</span></span>
<span id="cb7-98"><a href="#cb7-98" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="sc">-</span>Sample,</span>
<span id="cb7-99"><a href="#cb7-99" aria-hidden="true" tabindex="-1"></a>               <span class="at">names_to=</span><span class="st">"Genotype.ID"</span>,</span>
<span id="cb7-100"><a href="#cb7-100" aria-hidden="true" tabindex="-1"></a>               <span class="at">values_to=</span><span class="st">"Likelihood"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb7-101"><a href="#cb7-101" aria-hidden="true" tabindex="-1"></a>  <span class="fu">separate</span>(Genotype.ID, <span class="at">sep=</span><span class="dv">2</span>, <span class="at">into=</span><span class="fu">c</span>(<span class="st">"Allele"</span>,<span class="st">"Locus"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb7-102"><a href="#cb7-102" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Allele=</span><span class="fu">str_trunc</span>(Allele,<span class="dv">1</span>,<span class="at">side=</span><span class="st">"right"</span>, <span class="at">ellipsis=</span><span class="st">""</span>))</span>
<span id="cb7-103"><a href="#cb7-103" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-104"><a href="#cb7-104" aria-hidden="true" tabindex="-1"></a>df<span class="fl">.10</span> <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(genoprobs<span class="sc">$</span><span class="st">`</span><span class="at">10</span><span class="st">`</span>, <span class="at">stringsAsFactors =</span> <span class="cn">FALSE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb7-105"><a href="#cb7-105" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="at">cols =</span> <span class="fu">everything</span>(<span class="sc">-</span><span class="dv">1</span>),  <span class="co"># Replace "Sample" with the actual name of your sample identifier</span></span>
<span id="cb7-106"><a href="#cb7-106" aria-hidden="true" tabindex="-1"></a>               <span class="at">names_to =</span> <span class="st">"Sample"</span>,</span>
<span id="cb7-107"><a href="#cb7-107" aria-hidden="true" tabindex="-1"></a>               <span class="at">values_to =</span> <span class="st">"Value"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb7-108"><a href="#cb7-108" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Sample=</span><span class="fu">dimnames</span>(K<span class="sc">$</span>X)[[<span class="dv">1</span>]]) <span class="sc">%&gt;%</span> <span class="co">#added sample names which were lost in the previous step</span></span>
<span id="cb7-109"><a href="#cb7-109" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="sc">-</span>Sample,</span>
<span id="cb7-110"><a href="#cb7-110" aria-hidden="true" tabindex="-1"></a>               <span class="at">names_to=</span><span class="st">"Genotype.ID"</span>,</span>
<span id="cb7-111"><a href="#cb7-111" aria-hidden="true" tabindex="-1"></a>               <span class="at">values_to=</span><span class="st">"Likelihood"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb7-112"><a href="#cb7-112" aria-hidden="true" tabindex="-1"></a>  <span class="fu">separate</span>(Genotype.ID, <span class="at">sep=</span><span class="dv">2</span>, <span class="at">into=</span><span class="fu">c</span>(<span class="st">"Allele"</span>,<span class="st">"Locus"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb7-113"><a href="#cb7-113" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Allele=</span><span class="fu">str_trunc</span>(Allele,<span class="dv">1</span>,<span class="at">side=</span><span class="st">"right"</span>, <span class="at">ellipsis=</span><span class="st">""</span>))</span>
<span id="cb7-114"><a href="#cb7-114" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-115"><a href="#cb7-115" aria-hidden="true" tabindex="-1"></a>df<span class="fl">.11</span> <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(genoprobs<span class="sc">$</span><span class="st">`</span><span class="at">11</span><span class="st">`</span>, <span class="at">stringsAsFactors =</span> <span class="cn">FALSE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb7-116"><a href="#cb7-116" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="at">cols =</span> <span class="fu">everything</span>(<span class="sc">-</span><span class="dv">1</span>),  <span class="co"># Replace "Sample" with the actual name of your sample identifier</span></span>
<span id="cb7-117"><a href="#cb7-117" aria-hidden="true" tabindex="-1"></a>               <span class="at">names_to =</span> <span class="st">"Sample"</span>,</span>
<span id="cb7-118"><a href="#cb7-118" aria-hidden="true" tabindex="-1"></a>               <span class="at">values_to =</span> <span class="st">"Value"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb7-119"><a href="#cb7-119" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Sample=</span><span class="fu">dimnames</span>(K<span class="sc">$</span>X)[[<span class="dv">1</span>]]) <span class="sc">%&gt;%</span> <span class="co">#added sample names which were lost in the previous step</span></span>
<span id="cb7-120"><a href="#cb7-120" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="sc">-</span>Sample,</span>
<span id="cb7-121"><a href="#cb7-121" aria-hidden="true" tabindex="-1"></a>               <span class="at">names_to=</span><span class="st">"Genotype.ID"</span>,</span>
<span id="cb7-122"><a href="#cb7-122" aria-hidden="true" tabindex="-1"></a>               <span class="at">values_to=</span><span class="st">"Likelihood"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb7-123"><a href="#cb7-123" aria-hidden="true" tabindex="-1"></a>  <span class="fu">separate</span>(Genotype.ID, <span class="at">sep=</span><span class="dv">2</span>, <span class="at">into=</span><span class="fu">c</span>(<span class="st">"Allele"</span>,<span class="st">"Locus"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb7-124"><a href="#cb7-124" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Allele=</span><span class="fu">str_trunc</span>(Allele,<span class="dv">1</span>,<span class="at">side=</span><span class="st">"right"</span>, <span class="at">ellipsis=</span><span class="st">""</span>))</span>
<span id="cb7-125"><a href="#cb7-125" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-126"><a href="#cb7-126" aria-hidden="true" tabindex="-1"></a>df<span class="fl">.12</span> <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(genoprobs<span class="sc">$</span><span class="st">`</span><span class="at">12</span><span class="st">`</span>, <span class="at">stringsAsFactors =</span> <span class="cn">FALSE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb7-127"><a href="#cb7-127" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="at">cols =</span> <span class="fu">everything</span>(<span class="sc">-</span><span class="dv">1</span>),  <span class="co"># Replace "Sample" with the actual name of your sample identifier</span></span>
<span id="cb7-128"><a href="#cb7-128" aria-hidden="true" tabindex="-1"></a>               <span class="at">names_to =</span> <span class="st">"Sample"</span>,</span>
<span id="cb7-129"><a href="#cb7-129" aria-hidden="true" tabindex="-1"></a>               <span class="at">values_to =</span> <span class="st">"Value"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb7-130"><a href="#cb7-130" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Sample=</span><span class="fu">dimnames</span>(K<span class="sc">$</span>X)[[<span class="dv">1</span>]]) <span class="sc">%&gt;%</span> <span class="co">#added sample names which were lost in the previous step</span></span>
<span id="cb7-131"><a href="#cb7-131" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="sc">-</span>Sample,</span>
<span id="cb7-132"><a href="#cb7-132" aria-hidden="true" tabindex="-1"></a>               <span class="at">names_to=</span><span class="st">"Genotype.ID"</span>,</span>
<span id="cb7-133"><a href="#cb7-133" aria-hidden="true" tabindex="-1"></a>               <span class="at">values_to=</span><span class="st">"Likelihood"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb7-134"><a href="#cb7-134" aria-hidden="true" tabindex="-1"></a>  <span class="fu">separate</span>(Genotype.ID, <span class="at">sep=</span><span class="dv">2</span>, <span class="at">into=</span><span class="fu">c</span>(<span class="st">"Allele"</span>,<span class="st">"Locus"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb7-135"><a href="#cb7-135" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Allele=</span><span class="fu">str_trunc</span>(Allele,<span class="dv">1</span>,<span class="at">side=</span><span class="st">"right"</span>, <span class="at">ellipsis=</span><span class="st">""</span>))</span>
<span id="cb7-136"><a href="#cb7-136" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-137"><a href="#cb7-137" aria-hidden="true" tabindex="-1"></a>df<span class="fl">.13</span> <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(genoprobs<span class="sc">$</span><span class="st">`</span><span class="at">13</span><span class="st">`</span>, <span class="at">stringsAsFactors =</span> <span class="cn">FALSE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb7-138"><a href="#cb7-138" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="at">cols =</span> <span class="fu">everything</span>(<span class="sc">-</span><span class="dv">1</span>),  <span class="co"># Replace "Sample" with the actual name of your sample identifier</span></span>
<span id="cb7-139"><a href="#cb7-139" aria-hidden="true" tabindex="-1"></a>               <span class="at">names_to =</span> <span class="st">"Sample"</span>,</span>
<span id="cb7-140"><a href="#cb7-140" aria-hidden="true" tabindex="-1"></a>               <span class="at">values_to =</span> <span class="st">"Value"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb7-141"><a href="#cb7-141" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Sample=</span><span class="fu">dimnames</span>(K<span class="sc">$</span>X)[[<span class="dv">1</span>]]) <span class="sc">%&gt;%</span> <span class="co">#added sample names which were lost in the previous step</span></span>
<span id="cb7-142"><a href="#cb7-142" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="sc">-</span>Sample,</span>
<span id="cb7-143"><a href="#cb7-143" aria-hidden="true" tabindex="-1"></a>               <span class="at">names_to=</span><span class="st">"Genotype.ID"</span>,</span>
<span id="cb7-144"><a href="#cb7-144" aria-hidden="true" tabindex="-1"></a>               <span class="at">values_to=</span><span class="st">"Likelihood"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb7-145"><a href="#cb7-145" aria-hidden="true" tabindex="-1"></a>  <span class="fu">separate</span>(Genotype.ID, <span class="at">sep=</span><span class="dv">2</span>, <span class="at">into=</span><span class="fu">c</span>(<span class="st">"Allele"</span>,<span class="st">"Locus"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb7-146"><a href="#cb7-146" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Allele=</span><span class="fu">str_trunc</span>(Allele,<span class="dv">1</span>,<span class="at">side=</span><span class="st">"right"</span>, <span class="at">ellipsis=</span><span class="st">""</span>))</span>
<span id="cb7-147"><a href="#cb7-147" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-148"><a href="#cb7-148" aria-hidden="true" tabindex="-1"></a>df<span class="fl">.14</span> <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(genoprobs<span class="sc">$</span><span class="st">`</span><span class="at">14</span><span class="st">`</span>, <span class="at">stringsAsFactors =</span> <span class="cn">FALSE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb7-149"><a href="#cb7-149" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="at">cols =</span> <span class="fu">everything</span>(<span class="sc">-</span><span class="dv">1</span>),  <span class="co"># Replace "Sample" with the actual name of your sample identifier</span></span>
<span id="cb7-150"><a href="#cb7-150" aria-hidden="true" tabindex="-1"></a>               <span class="at">names_to =</span> <span class="st">"Sample"</span>,</span>
<span id="cb7-151"><a href="#cb7-151" aria-hidden="true" tabindex="-1"></a>               <span class="at">values_to =</span> <span class="st">"Value"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb7-152"><a href="#cb7-152" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Sample=</span><span class="fu">dimnames</span>(K<span class="sc">$</span>X)[[<span class="dv">1</span>]]) <span class="sc">%&gt;%</span> <span class="co">#added sample names which were lost in the previous step</span></span>
<span id="cb7-153"><a href="#cb7-153" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="sc">-</span>Sample,</span>
<span id="cb7-154"><a href="#cb7-154" aria-hidden="true" tabindex="-1"></a>               <span class="at">names_to=</span><span class="st">"Genotype.ID"</span>,</span>
<span id="cb7-155"><a href="#cb7-155" aria-hidden="true" tabindex="-1"></a>               <span class="at">values_to=</span><span class="st">"Likelihood"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb7-156"><a href="#cb7-156" aria-hidden="true" tabindex="-1"></a>  <span class="fu">separate</span>(Genotype.ID, <span class="at">sep=</span><span class="dv">2</span>, <span class="at">into=</span><span class="fu">c</span>(<span class="st">"Allele"</span>,<span class="st">"Locus"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb7-157"><a href="#cb7-157" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Allele=</span><span class="fu">str_trunc</span>(Allele,<span class="dv">1</span>,<span class="at">side=</span><span class="st">"right"</span>, <span class="at">ellipsis=</span><span class="st">""</span>))</span>
<span id="cb7-158"><a href="#cb7-158" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-159"><a href="#cb7-159" aria-hidden="true" tabindex="-1"></a>df<span class="fl">.15</span> <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(genoprobs<span class="sc">$</span><span class="st">`</span><span class="at">15</span><span class="st">`</span>, <span class="at">stringsAsFactors =</span> <span class="cn">FALSE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb7-160"><a href="#cb7-160" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="at">cols =</span> <span class="fu">everything</span>(<span class="sc">-</span><span class="dv">1</span>),  <span class="co"># Replace "Sample" with the actual name of your sample identifier</span></span>
<span id="cb7-161"><a href="#cb7-161" aria-hidden="true" tabindex="-1"></a>               <span class="at">names_to =</span> <span class="st">"Sample"</span>,</span>
<span id="cb7-162"><a href="#cb7-162" aria-hidden="true" tabindex="-1"></a>               <span class="at">values_to =</span> <span class="st">"Value"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb7-163"><a href="#cb7-163" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Sample=</span><span class="fu">dimnames</span>(K<span class="sc">$</span>X)[[<span class="dv">1</span>]]) <span class="sc">%&gt;%</span> <span class="co">#added sample names which were lost in the previous step</span></span>
<span id="cb7-164"><a href="#cb7-164" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="sc">-</span>Sample,</span>
<span id="cb7-165"><a href="#cb7-165" aria-hidden="true" tabindex="-1"></a>               <span class="at">names_to=</span><span class="st">"Genotype.ID"</span>,</span>
<span id="cb7-166"><a href="#cb7-166" aria-hidden="true" tabindex="-1"></a>               <span class="at">values_to=</span><span class="st">"Likelihood"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb7-167"><a href="#cb7-167" aria-hidden="true" tabindex="-1"></a>  <span class="fu">separate</span>(Genotype.ID, <span class="at">sep=</span><span class="dv">2</span>, <span class="at">into=</span><span class="fu">c</span>(<span class="st">"Allele"</span>,<span class="st">"Locus"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb7-168"><a href="#cb7-168" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Allele=</span><span class="fu">str_trunc</span>(Allele,<span class="dv">1</span>,<span class="at">side=</span><span class="st">"right"</span>, <span class="at">ellipsis=</span><span class="st">""</span>))</span>
<span id="cb7-169"><a href="#cb7-169" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-170"><a href="#cb7-170" aria-hidden="true" tabindex="-1"></a>df<span class="fl">.16</span> <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(genoprobs<span class="sc">$</span><span class="st">`</span><span class="at">16</span><span class="st">`</span>, <span class="at">stringsAsFactors =</span> <span class="cn">FALSE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb7-171"><a href="#cb7-171" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="at">cols =</span> <span class="fu">everything</span>(<span class="sc">-</span><span class="dv">1</span>),  <span class="co"># Replace "Sample" with the actual name of your sample identifier</span></span>
<span id="cb7-172"><a href="#cb7-172" aria-hidden="true" tabindex="-1"></a>               <span class="at">names_to =</span> <span class="st">"Sample"</span>,</span>
<span id="cb7-173"><a href="#cb7-173" aria-hidden="true" tabindex="-1"></a>               <span class="at">values_to =</span> <span class="st">"Value"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb7-174"><a href="#cb7-174" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Sample=</span><span class="fu">dimnames</span>(K<span class="sc">$</span>X)[[<span class="dv">1</span>]]) <span class="sc">%&gt;%</span> <span class="co">#added sample names which were lost in the previous step</span></span>
<span id="cb7-175"><a href="#cb7-175" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="sc">-</span>Sample,</span>
<span id="cb7-176"><a href="#cb7-176" aria-hidden="true" tabindex="-1"></a>               <span class="at">names_to=</span><span class="st">"Genotype.ID"</span>,</span>
<span id="cb7-177"><a href="#cb7-177" aria-hidden="true" tabindex="-1"></a>               <span class="at">values_to=</span><span class="st">"Likelihood"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb7-178"><a href="#cb7-178" aria-hidden="true" tabindex="-1"></a>  <span class="fu">separate</span>(Genotype.ID, <span class="at">sep=</span><span class="dv">2</span>, <span class="at">into=</span><span class="fu">c</span>(<span class="st">"Allele"</span>,<span class="st">"Locus"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb7-179"><a href="#cb7-179" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Allele=</span><span class="fu">str_trunc</span>(Allele,<span class="dv">1</span>,<span class="at">side=</span><span class="st">"right"</span>, <span class="at">ellipsis=</span><span class="st">""</span>))</span>
<span id="cb7-180"><a href="#cb7-180" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-181"><a href="#cb7-181" aria-hidden="true" tabindex="-1"></a>df<span class="fl">.16</span> <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(genoprobs<span class="sc">$</span><span class="st">`</span><span class="at">16</span><span class="st">`</span>, <span class="at">stringsAsFactors =</span> <span class="cn">FALSE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb7-182"><a href="#cb7-182" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="at">cols =</span> <span class="fu">everything</span>(<span class="sc">-</span><span class="dv">1</span>),  <span class="co"># Replace "Sample" with the actual name of your sample identifier</span></span>
<span id="cb7-183"><a href="#cb7-183" aria-hidden="true" tabindex="-1"></a>               <span class="at">names_to =</span> <span class="st">"Sample"</span>,</span>
<span id="cb7-184"><a href="#cb7-184" aria-hidden="true" tabindex="-1"></a>               <span class="at">values_to =</span> <span class="st">"Value"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb7-185"><a href="#cb7-185" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Sample=</span><span class="fu">dimnames</span>(K<span class="sc">$</span>X)[[<span class="dv">1</span>]]) <span class="sc">%&gt;%</span> <span class="co">#added sample names which were lost in the previous step</span></span>
<span id="cb7-186"><a href="#cb7-186" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="sc">-</span>Sample,</span>
<span id="cb7-187"><a href="#cb7-187" aria-hidden="true" tabindex="-1"></a>               <span class="at">names_to=</span><span class="st">"Genotype.ID"</span>,</span>
<span id="cb7-188"><a href="#cb7-188" aria-hidden="true" tabindex="-1"></a>               <span class="at">values_to=</span><span class="st">"Likelihood"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb7-189"><a href="#cb7-189" aria-hidden="true" tabindex="-1"></a>  <span class="fu">separate</span>(Genotype.ID, <span class="at">sep=</span><span class="dv">2</span>, <span class="at">into=</span><span class="fu">c</span>(<span class="st">"Allele"</span>,<span class="st">"Locus"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb7-190"><a href="#cb7-190" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Allele=</span><span class="fu">str_trunc</span>(Allele,<span class="dv">1</span>,<span class="at">side=</span><span class="st">"right"</span>, <span class="at">ellipsis=</span><span class="st">""</span>))</span>
<span id="cb7-191"><a href="#cb7-191" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-192"><a href="#cb7-192" aria-hidden="true" tabindex="-1"></a>df<span class="fl">.17</span> <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(genoprobs<span class="sc">$</span><span class="st">`</span><span class="at">17</span><span class="st">`</span>, <span class="at">stringsAsFactors =</span> <span class="cn">FALSE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb7-193"><a href="#cb7-193" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="at">cols =</span> <span class="fu">everything</span>(<span class="sc">-</span><span class="dv">1</span>),  <span class="co"># Replace "Sample" with the actual name of your sample identifier</span></span>
<span id="cb7-194"><a href="#cb7-194" aria-hidden="true" tabindex="-1"></a>               <span class="at">names_to =</span> <span class="st">"Sample"</span>,</span>
<span id="cb7-195"><a href="#cb7-195" aria-hidden="true" tabindex="-1"></a>               <span class="at">values_to =</span> <span class="st">"Value"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb7-196"><a href="#cb7-196" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Sample=</span><span class="fu">dimnames</span>(K<span class="sc">$</span>X)[[<span class="dv">1</span>]]) <span class="sc">%&gt;%</span> <span class="co">#added sample names which were lost in the previous step</span></span>
<span id="cb7-197"><a href="#cb7-197" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="sc">-</span>Sample,</span>
<span id="cb7-198"><a href="#cb7-198" aria-hidden="true" tabindex="-1"></a>               <span class="at">names_to=</span><span class="st">"Genotype.ID"</span>,</span>
<span id="cb7-199"><a href="#cb7-199" aria-hidden="true" tabindex="-1"></a>               <span class="at">values_to=</span><span class="st">"Likelihood"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb7-200"><a href="#cb7-200" aria-hidden="true" tabindex="-1"></a>  <span class="fu">separate</span>(Genotype.ID, <span class="at">sep=</span><span class="dv">2</span>, <span class="at">into=</span><span class="fu">c</span>(<span class="st">"Allele"</span>,<span class="st">"Locus"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb7-201"><a href="#cb7-201" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Allele=</span><span class="fu">str_trunc</span>(Allele,<span class="dv">1</span>,<span class="at">side=</span><span class="st">"right"</span>, <span class="at">ellipsis=</span><span class="st">""</span>))</span>
<span id="cb7-202"><a href="#cb7-202" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-203"><a href="#cb7-203" aria-hidden="true" tabindex="-1"></a>df<span class="fl">.18</span> <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(genoprobs<span class="sc">$</span><span class="st">`</span><span class="at">18</span><span class="st">`</span>, <span class="at">stringsAsFactors =</span> <span class="cn">FALSE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb7-204"><a href="#cb7-204" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="at">cols =</span> <span class="fu">everything</span>(<span class="sc">-</span><span class="dv">1</span>),  <span class="co"># Replace "Sample" with the actual name of your sample identifier</span></span>
<span id="cb7-205"><a href="#cb7-205" aria-hidden="true" tabindex="-1"></a>               <span class="at">names_to =</span> <span class="st">"Sample"</span>,</span>
<span id="cb7-206"><a href="#cb7-206" aria-hidden="true" tabindex="-1"></a>               <span class="at">values_to =</span> <span class="st">"Value"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb7-207"><a href="#cb7-207" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Sample=</span><span class="fu">dimnames</span>(K<span class="sc">$</span>X)[[<span class="dv">1</span>]]) <span class="sc">%&gt;%</span> <span class="co">#added sample names which were lost in the previous step</span></span>
<span id="cb7-208"><a href="#cb7-208" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="sc">-</span>Sample,</span>
<span id="cb7-209"><a href="#cb7-209" aria-hidden="true" tabindex="-1"></a>               <span class="at">names_to=</span><span class="st">"Genotype.ID"</span>,</span>
<span id="cb7-210"><a href="#cb7-210" aria-hidden="true" tabindex="-1"></a>               <span class="at">values_to=</span><span class="st">"Likelihood"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb7-211"><a href="#cb7-211" aria-hidden="true" tabindex="-1"></a>  <span class="fu">separate</span>(Genotype.ID, <span class="at">sep=</span><span class="dv">2</span>, <span class="at">into=</span><span class="fu">c</span>(<span class="st">"Allele"</span>,<span class="st">"Locus"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb7-212"><a href="#cb7-212" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Allele=</span><span class="fu">str_trunc</span>(Allele,<span class="dv">1</span>,<span class="at">side=</span><span class="st">"right"</span>, <span class="at">ellipsis=</span><span class="st">""</span>))</span>
<span id="cb7-213"><a href="#cb7-213" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-214"><a href="#cb7-214" aria-hidden="true" tabindex="-1"></a>df<span class="fl">.19</span> <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(genoprobs<span class="sc">$</span><span class="st">`</span><span class="at">19</span><span class="st">`</span>, <span class="at">stringsAsFactors =</span> <span class="cn">FALSE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb7-215"><a href="#cb7-215" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="at">cols =</span> <span class="fu">everything</span>(<span class="sc">-</span><span class="dv">1</span>),  <span class="co"># Replace "Sample" with the actual name of your sample identifier</span></span>
<span id="cb7-216"><a href="#cb7-216" aria-hidden="true" tabindex="-1"></a>               <span class="at">names_to =</span> <span class="st">"Sample"</span>,</span>
<span id="cb7-217"><a href="#cb7-217" aria-hidden="true" tabindex="-1"></a>               <span class="at">values_to =</span> <span class="st">"Value"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb7-218"><a href="#cb7-218" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Sample=</span><span class="fu">dimnames</span>(K<span class="sc">$</span>X)[[<span class="dv">1</span>]]) <span class="sc">%&gt;%</span> <span class="co">#added sample names which were lost in the previous step</span></span>
<span id="cb7-219"><a href="#cb7-219" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="sc">-</span>Sample,</span>
<span id="cb7-220"><a href="#cb7-220" aria-hidden="true" tabindex="-1"></a>               <span class="at">names_to=</span><span class="st">"Genotype.ID"</span>,</span>
<span id="cb7-221"><a href="#cb7-221" aria-hidden="true" tabindex="-1"></a>               <span class="at">values_to=</span><span class="st">"Likelihood"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb7-222"><a href="#cb7-222" aria-hidden="true" tabindex="-1"></a>  <span class="fu">separate</span>(Genotype.ID, <span class="at">sep=</span><span class="dv">2</span>, <span class="at">into=</span><span class="fu">c</span>(<span class="st">"Allele"</span>,<span class="st">"Locus"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb7-223"><a href="#cb7-223" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Allele=</span><span class="fu">str_trunc</span>(Allele,<span class="dv">1</span>,<span class="at">side=</span><span class="st">"right"</span>, <span class="at">ellipsis=</span><span class="st">""</span>))</span>
<span id="cb7-224"><a href="#cb7-224" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-225"><a href="#cb7-225" aria-hidden="true" tabindex="-1"></a>df.X <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(genoprobs<span class="sc">$</span><span class="st">`</span><span class="at">X</span><span class="st">`</span>, <span class="at">stringsAsFactors =</span> <span class="cn">FALSE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb7-226"><a href="#cb7-226" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="at">cols =</span> <span class="fu">everything</span>(<span class="sc">-</span><span class="dv">1</span>),  <span class="co"># Replace "Sample" with the actual name of your sample identifier</span></span>
<span id="cb7-227"><a href="#cb7-227" aria-hidden="true" tabindex="-1"></a>               <span class="at">names_to =</span> <span class="st">"Sample"</span>,</span>
<span id="cb7-228"><a href="#cb7-228" aria-hidden="true" tabindex="-1"></a>               <span class="at">values_to =</span> <span class="st">"Value"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb7-229"><a href="#cb7-229" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Sample=</span><span class="fu">dimnames</span>(K<span class="sc">$</span>X)[[<span class="dv">1</span>]]) <span class="sc">%&gt;%</span> <span class="co">#added sample names which were lost in the previous step</span></span>
<span id="cb7-230"><a href="#cb7-230" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="sc">-</span>Sample,</span>
<span id="cb7-231"><a href="#cb7-231" aria-hidden="true" tabindex="-1"></a>               <span class="at">names_to=</span><span class="st">"Genotype.ID"</span>,</span>
<span id="cb7-232"><a href="#cb7-232" aria-hidden="true" tabindex="-1"></a>               <span class="at">values_to=</span><span class="st">"Likelihood"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb7-233"><a href="#cb7-233" aria-hidden="true" tabindex="-1"></a>  <span class="fu">separate</span>(Genotype.ID, <span class="at">sep=</span><span class="dv">2</span>, <span class="at">into=</span><span class="fu">c</span>(<span class="st">"Allele"</span>,<span class="st">"Locus"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb7-234"><a href="#cb7-234" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Allele=</span><span class="fu">str_trunc</span>(Allele,<span class="dv">1</span>,<span class="at">side=</span><span class="st">"right"</span>, <span class="at">ellipsis=</span><span class="st">""</span>))</span>
<span id="cb7-235"><a href="#cb7-235" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-236"><a href="#cb7-236" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-237"><a href="#cb7-237" aria-hidden="true" tabindex="-1"></a>long_df <span class="ot">&lt;-</span> <span class="fu">bind_rows</span>(df<span class="fl">.1</span>,</span>
<span id="cb7-238"><a href="#cb7-238" aria-hidden="true" tabindex="-1"></a>                     df<span class="fl">.2</span>,</span>
<span id="cb7-239"><a href="#cb7-239" aria-hidden="true" tabindex="-1"></a>                     df<span class="fl">.3</span>,</span>
<span id="cb7-240"><a href="#cb7-240" aria-hidden="true" tabindex="-1"></a>                     df<span class="fl">.4</span>,</span>
<span id="cb7-241"><a href="#cb7-241" aria-hidden="true" tabindex="-1"></a>                     df<span class="fl">.5</span>,</span>
<span id="cb7-242"><a href="#cb7-242" aria-hidden="true" tabindex="-1"></a>                     df<span class="fl">.6</span>,</span>
<span id="cb7-243"><a href="#cb7-243" aria-hidden="true" tabindex="-1"></a>                     df<span class="fl">.7</span>,</span>
<span id="cb7-244"><a href="#cb7-244" aria-hidden="true" tabindex="-1"></a>                     df<span class="fl">.8</span>,</span>
<span id="cb7-245"><a href="#cb7-245" aria-hidden="true" tabindex="-1"></a>                     df<span class="fl">.9</span>,</span>
<span id="cb7-246"><a href="#cb7-246" aria-hidden="true" tabindex="-1"></a>                     df<span class="fl">.10</span>,</span>
<span id="cb7-247"><a href="#cb7-247" aria-hidden="true" tabindex="-1"></a>                     df<span class="fl">.11</span>,</span>
<span id="cb7-248"><a href="#cb7-248" aria-hidden="true" tabindex="-1"></a>                     df<span class="fl">.12</span>,</span>
<span id="cb7-249"><a href="#cb7-249" aria-hidden="true" tabindex="-1"></a>                     df<span class="fl">.13</span>,</span>
<span id="cb7-250"><a href="#cb7-250" aria-hidden="true" tabindex="-1"></a>                     df<span class="fl">.14</span>,</span>
<span id="cb7-251"><a href="#cb7-251" aria-hidden="true" tabindex="-1"></a>                     df<span class="fl">.15</span>,</span>
<span id="cb7-252"><a href="#cb7-252" aria-hidden="true" tabindex="-1"></a>                     df<span class="fl">.16</span>,</span>
<span id="cb7-253"><a href="#cb7-253" aria-hidden="true" tabindex="-1"></a>                     df<span class="fl">.17</span>,</span>
<span id="cb7-254"><a href="#cb7-254" aria-hidden="true" tabindex="-1"></a>                     df<span class="fl">.18</span>,</span>
<span id="cb7-255"><a href="#cb7-255" aria-hidden="true" tabindex="-1"></a>                     df<span class="fl">.19</span>,</span>
<span id="cb7-256"><a href="#cb7-256" aria-hidden="true" tabindex="-1"></a>                     df.X)</span>
<span id="cb7-257"><a href="#cb7-257" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-258"><a href="#cb7-258" aria-hidden="true" tabindex="-1"></a><span class="co">#now need to find out the major allele for each locus</span></span>
<span id="cb7-259"><a href="#cb7-259" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-260"><a href="#cb7-260" aria-hidden="true" tabindex="-1"></a>major_allele <span class="ot">&lt;-</span></span>
<span id="cb7-261"><a href="#cb7-261" aria-hidden="true" tabindex="-1"></a>  long_df <span class="sc">%&gt;%</span></span>
<span id="cb7-262"><a href="#cb7-262" aria-hidden="true" tabindex="-1"></a>  ungroup <span class="sc">%&gt;%</span></span>
<span id="cb7-263"><a href="#cb7-263" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(Locus) <span class="sc">%&gt;%</span></span>
<span id="cb7-264"><a href="#cb7-264" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Likelihood <span class="sc">==</span> <span class="fu">max</span>(Likelihood)) <span class="sc">%&gt;%</span> <span class="co">#select only the allele which is the major allele for a locus </span></span>
<span id="cb7-265"><a href="#cb7-265" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="sc">-</span>Sample)</span>
<span id="cb7-266"><a href="#cb7-266" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-267"><a href="#cb7-267" aria-hidden="true" tabindex="-1"></a>knitr<span class="sc">::</span><span class="fu">kable</span>(<span class="fu">as.data.frame</span>(<span class="fu">summary</span>(<span class="fu">as.factor</span>(major_allele<span class="sc">$</span>Allele))) <span class="sc">%&gt;%</span></span>
<span id="cb7-268"><a href="#cb7-268" aria-hidden="true" tabindex="-1"></a>               <span class="fu">rename</span>(<span class="at">Count=</span><span class="dv">1</span>) <span class="sc">%&gt;%</span></span>
<span id="cb7-269"><a href="#cb7-269" aria-hidden="true" tabindex="-1"></a>               <span class="fu">mutate</span>(<span class="at">Pct =</span> Count<span class="sc">/</span><span class="fu">sum</span>(Count)<span class="sc">*</span><span class="dv">100</span>),</span>
<span id="cb7-270"><a href="#cb7-270" aria-hidden="true" tabindex="-1"></a>      <span class="at">caption=</span><span class="st">"Number of major alleles per originating strain"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<table class="table table-sm table-striped small">
<caption>Number of major alleles per originating strain</caption>
<thead>
<tr class="header">
<th style="text-align: left;"></th>
<th style="text-align: right;">Count</th>
<th style="text-align: right;">Pct</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">A</td>
<td style="text-align: right;">5094</td>
<td style="text-align: right;">7.1754564</td>
</tr>
<tr class="even">
<td style="text-align: left;">B</td>
<td style="text-align: right;">7664</td>
<td style="text-align: right;">10.7955826</td>
</tr>
<tr class="odd">
<td style="text-align: left;">C</td>
<td style="text-align: right;">8616</td>
<td style="text-align: right;">12.1365788</td>
</tr>
<tr class="even">
<td style="text-align: left;">D</td>
<td style="text-align: right;">6598</td>
<td style="text-align: right;">9.2940050</td>
</tr>
<tr class="odd">
<td style="text-align: left;">E</td>
<td style="text-align: right;">6379</td>
<td style="text-align: right;">8.9855195</td>
</tr>
<tr class="even">
<td style="text-align: left;">F</td>
<td style="text-align: right;">13817</td>
<td style="text-align: right;">19.4627564</td>
</tr>
<tr class="odd">
<td style="text-align: left;">G</td>
<td style="text-align: right;">9358</td>
<td style="text-align: right;">13.1817670</td>
</tr>
<tr class="even">
<td style="text-align: left;">H</td>
<td style="text-align: right;">13465</td>
<td style="text-align: right;">18.9669259</td>
</tr>
<tr class="odd">
<td style="text-align: left;">V</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">0.0014086</td>
</tr>
</tbody>
</table>
</div>
<div class="sourceCode cell-code" id="cb8"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a>long.data.major <span class="ot">&lt;-</span></span>
<span id="cb8-2"><a href="#cb8-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(long_df,major_allele, <span class="at">by=</span><span class="st">"Locus"</span>, <span class="at">suffix=</span><span class="fu">c</span>(<span class="st">""</span>,<span class="st">"_major"</span>)) <span class="sc">%&gt;%</span> <span class="co">#merged in to have the major allele for each sample/snp pair</span></span>
<span id="cb8-3"><a href="#cb8-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">Major=</span><span class="st">"Allele_major"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb8-4"><a href="#cb8-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="sc">-</span>Likelihood_major) <span class="sc">%&gt;%</span> <span class="co">#not used further</span></span>
<span id="cb8-5"><a href="#cb8-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Likelihood<span class="sc">&gt;</span><span class="fl">0.1</span>)<span class="co">#remove variants where the predicted allele likelihood is very low.  Ideally this should be 0.5, 0 or 1</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in left_join(long_df, major_allele, by = "Locus", suffix = c("", : Detected an unexpected many-to-many relationship between `x` and `y`.
ℹ Row 9600 of `x` matches multiple rows in `y`.
ℹ Row 3175 of `y` matches multiple rows in `x`.
ℹ If a many-to-many relationship is expected, set `relationship =
  "many-to-many"` to silence this warning.</code></pre>
</div>
<div class="sourceCode cell-code" id="cb10"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a>bimbam.format <span class="ot">&lt;-</span></span>
<span id="cb10-2"><a href="#cb10-2" aria-hidden="true" tabindex="-1"></a>  long.data.major <span class="sc">%&gt;%</span></span>
<span id="cb10-3"><a href="#cb10-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Locus.Combined=</span><span class="fu">paste</span>(Locus,Major,Allele,<span class="at">sep=</span><span class="st">"_"</span>)) <span class="sc">%&gt;%</span> <span class="co">#renamed locus to be CHR_LOC_MAJ_MIN, these are new genotypes because any given locus can have multiple variants</span></span>
<span id="cb10-4"><a href="#cb10-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(Locus.Combined, Sample, Likelihood) <span class="sc">%&gt;%</span></span>
<span id="cb10-5"><a href="#cb10-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_wider</span>(<span class="at">names_from=</span><span class="st">"Sample"</span>,</span>
<span id="cb10-6"><a href="#cb10-6" aria-hidden="true" tabindex="-1"></a>              <span class="at">values_from=</span><span class="st">"Likelihood"</span>,</span>
<span id="cb10-7"><a href="#cb10-7" aria-hidden="true" tabindex="-1"></a>              <span class="at">values_fn=</span>mean) <span class="sc">%&gt;%</span> </span>
<span id="cb10-8"><a href="#cb10-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate_all</span>(<span class="sc">~</span><span class="fu">replace</span>(., <span class="fu">is.na</span>(.), <span class="dv">0</span>)) <span class="sc">%&gt;%</span> <span class="co">#convert NA values to zero</span></span>
<span id="cb10-9"><a href="#cb10-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate_if</span>(is.double,<span class="at">.funs=</span><span class="sc">~</span>.<span class="sc">*</span><span class="dv">2</span>)  <span class="co">#multiplied by 2 to make heterozygosity a 1</span></span>
<span id="cb10-10"><a href="#cb10-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-11"><a href="#cb10-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-12"><a href="#cb10-12" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(readr)</span>
<span id="cb10-13"><a href="#cb10-13" aria-hidden="true" tabindex="-1"></a><span class="co">#write out the genome file in the right format</span></span>
<span id="cb10-14"><a href="#cb10-14" aria-hidden="true" tabindex="-1"></a>genotype.file <span class="ot">&lt;-</span> <span class="st">"DO_HFD_Genome_All.csv"</span></span>
<span id="cb10-15"><a href="#cb10-15" aria-hidden="true" tabindex="-1"></a><span class="fu">write_csv</span>(bimbam.format, <span class="at">file=</span>genotype.file)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>This script creates a csv file which contains both annotation data and the genotype data called <code>r</code>. This is used by the <em>SNP-Annotation.qmd</em> file to parse out the genotypes from the SNP annotaiton.</p>
</section>
<section id="session-information" class="level1">
<h1>Session Information</h1>
<div class="cell">
<div class="sourceCode cell-code" id="cb11"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a><span class="fu">sessionInfo</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>R version 4.3.1 (2023-06-16)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Red Hat Enterprise Linux 8.6 (Ootpa)

Matrix products: default
BLAS:   /sw/pkgs/arc/stacks/gcc/10.3.0/R/4.3.1/lib64/R/lib/libRblas.so 
LAPACK: /sw/pkgs/arc/stacks/gcc/10.3.0/R/4.3.1/lib64/R/lib/libRlapack.so;  LAPACK version 3.11.0

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

time zone: America/Detroit
tzcode source: system (glibc)

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] readr_2.1.4   stringr_1.5.0 tibble_3.2.1  dplyr_1.1.3   tidyr_1.3.0  

loaded via a namespace (and not attached):
 [1] crayon_1.5.2     vctrs_0.6.4      cli_3.6.1        knitr_1.45      
 [5] rlang_1.1.1      xfun_0.41        stringi_1.7.12   purrr_1.0.2     
 [9] generics_0.1.3   jsonlite_1.8.7   bit_4.0.5        glue_1.6.2      
[13] htmltools_0.5.7  hms_1.1.3        fansi_1.0.5      rmarkdown_2.25  
[17] evaluate_0.23    tzdb_0.4.0       fastmap_1.1.1    yaml_2.3.7      
[21] lifecycle_1.0.3  compiler_4.3.1   pkgconfig_2.0.3  digest_0.6.33   
[25] R6_2.5.1         tidyselect_1.2.0 utf8_1.2.4       parallel_4.3.1  
[29] vroom_1.6.4      pillar_1.9.0     magrittr_2.0.3   bit64_4.0.5     
[33] tools_4.3.1      withr_2.5.2     </code></pre>
</div>
</div>
</section>

</main>
<!-- /main column -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->



</body></html>