---
title: "Creating Phenotype Files from HFD-DO Viewer Dataset"
author: "Dave Bridges"
format: html
---

# Load Dataset

```{r load}
library(readr)
library(dplyr)
phenotype.datatfile <- 'dataset.phenotype.Svenson_DO_HFD.v12.Rds'
phenotype.data <- readRDS(phenotype.datatfile)

genotype.file <- 'DO_HFD_Genome_All.csv'
genotype.data <- read_csv(genotype.file)
```

# Phenotype Annotation file format

looks like this (from https://www.xzlab.org/software/GEMMAmanual.pdf)

This is a single column of phenotypes, in the order of the genotypes



```{r snp-annotation}
library(tidyr)

chol.phenotypes <- 
  phenotype.data$data$raw %>% 
  as.data.frame %>%
  select(chol2)

strains.in.order <- 
  genotype.data %>%
  dplyr::select(-Locus.Combined) %>%
  colnames

chol.phenotypes[strains.in.order,] %>% write_lines(file="Cholesterol_all.txt")
```

This file should be usable for all GEMMA runs that contain all the mice

## Plink PED and FAM Files

See https://zzz.bwh.harvard.edu/plink/data.shtml#ped

```{r plink-files}
chol.phenotypes.pl <- 
  phenotype.data$data$raw %>% 
  as.data.frame %>%
  select(chol2) %>%
  tibble::rownames_to_column("ID") %>%
  separate(ID, sep=1, into=c("SexName","Number"), remove=F) %>%
  mutate(Family=paste0("Family",row_number()),
         Paternal=0,
         Maternal=0) %>%
  mutate(Sex=case_when(SexName=="F"~'2',
                       SexName=="M"~'1',
                       .default="other")) %>%
  mutate(Chol=ifelse(is.na(chol2), -9, chol2)) %>%
  mutate(ID2 = ID) %>%
  tibble::column_to_rownames("ID2")

chol.phenotypes.pl[strains.in.order,c("Family","ID","Paternal","Maternal","Sex","chol2")] %>% 
  write_tsv(file="Genotypes_all.ped", col_names = F)

#same format for fam files
chol.phenotypes.pl[strains.in.order,c("Family","ID","Paternal","Maternal","Sex","chol2")] %>% 
  write_tsv(file="Genotypes_all.fam", col_names = F)

```

# Covariate Annotation File

Needed covariates for sex and diet

```{r covariates}
library(tibble)
library(forcats)
phenotype.data$annot.samples %>%
  select(mouse.id,sex,diet) %>% 
  column_to_rownames('mouse.id') %>% 
  mutate_all(as.factor) %>%
  mutate(intercept=1,
         cov.sex=case_when(sex=='F'~1,
                       sex=='M'~0),
         cov.diet=case_when(diet=='chow'~0,
                       diet=='hf'~1)) %>% 
  select(-sex,-diet) -> covariates.data 

covariates.data[strains.in.order,] %>% #mandates covariates are in same order, adds na's if necessary
  write_delim(delim=" ",
              col_names = F,
              file="Covariates.tab") %>%
  select(-cov.diet) %>%
  write_delim(delim=" ",
              col_names=F,
              file="Covariates.sex.tab")

    
#wrote out covariates file with diet as a covariate (ie removed sex covariate)
covariates.data[strains.in.order,] %>%  
  select(-cov.sex) %>%
    write_delim(delim=" ",
                col_names=F,
                file="Covariates.diet.tab")
           
```

## Covariates and Environment File for GxE Analysis

```{r gxe-files}

covariates.data[strains.in.order,c('intercept','cov.sex')] %>% #mandates covariates are in same order, adds na's if necessary
  write_delim(delim=" ",
              col_names = F,
              file="Covariate.gxe.tab")

covariates.data[strains.in.order,] %>% #mandates covariates are in same order, adds na's if necessary
  pull(cov.diet) %>%
  write(file="Diet.gxe.tab",ncolumns=1)
```

For the covariates file 0 is the chow diet and male sex

# Diet Specific Phenotype Files

Marked phenotypes as NA when not needed so that the same genotype files can be used.  Need to make new covariates file

## For Chow Only

```{r phenotypes-ncd}
ncd.mice <- rownames(filter(covariates.data,cov.diet=='0'))

ncd.cholesterol <-
  chol.phenotypes %>%
  mutate(chol=ifelse(rownames(.) %in% ncd.mice,chol2,NA)) %>%
  select(chol)
         
ncd.cholesterol[strains.in.order,] %>% write_lines(file="Cholesterol_ncd.txt")
```

## For HFD Only

```{r phenotypes-hfd}
hfd.mice <- rownames(filter(covariates.data,cov.diet=='1'))

hfd.cholesterol <-
  chol.phenotypes %>%
  mutate(chol=ifelse(rownames(.) %in% hfd.mice,chol2,NA)) %>%
  select(chol)
         
hfd.cholesterol[strains.in.order,] %>% write_lines(file="Cholesterol_hfd.txt")
```

# Sex Specific Phenotype Files

Marked phenotypes as NA when not needed so that the same genotype files can be used.  Need to make new covariates file

## For Males Only

```{r phenotypes-male}
male.mice <- rownames(filter(covariates.data,cov.sex=='0'))

male.cholesterol <-
  chol.phenotypes %>%
  mutate(chol=ifelse(rownames(.) %in% ncd.mice,chol2,NA)) %>%
  select(chol)
         
male.cholesterol[strains.in.order,] %>% write_lines(file="Cholesterol_male.txt")
```

## For Female Only

```{r phenotypes-female}
female.mice <- rownames(filter(covariates.data,cov.sex=='1'))

female.cholesterol <-
  chol.phenotypes %>%
  mutate(chol=ifelse(rownames(.) %in% hfd.mice,chol2,NA)) %>%
  select(chol)
         
female.cholesterol[strains.in.order,] %>% write_lines(file="Cholesterol_female.txt")
```

# Diet and Sex Adjusted Cholesterol

This is needed for BSLMM which does not use a covariates file

```{r phenotypes-adjusted}
merged.data <- merge(covariates.data,chol.phenotypes,by='row.names')

additive.model <- lm(chol2~cov.sex + cov.diet,data=merged.data)
library(broom)
library(knitr)
summary(additive.model) %>% tidy %>% kable(caption="Summary of additive model")

chol.adj.data <-
  merged.data %>%
  mutate(chol.adj = case_when(
    cov.sex==0&cov.diet==0~chol2,
    cov.sex==1&cov.diet==0~chol2+coefficients(additive.model)['cov.sex'],
    cov.sex==0&cov.diet==1~chol2+coefficients(additive.model)['cov.diet'],
    cov.sex==1&cov.diet==1~chol2+coefficients(additive.model)['cov.sex']+coefficients(additive.model)['cov.diet'])) %>%
  column_to_rownames('Row.names') %>%
  select(chol.adj)

chol.adj.data[strains.in.order,] %>% write_lines(file="Cholesterol_adj.txt")
```


# Session Information

```{r session-info}
sessionInfo()
```
